<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Function reference â€¢ scMultiSim</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scMultiSim</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/workflow.html">Basic Workflow</a>
    <a class="dropdown-item" href="../articles/basics.html">Simulating Multimodal Single-cell Datasets</a>
    <a class="dropdown-item" href="../articles/spatialCCI.html">Simulating Spatial Cell-Cell Interactions</a>
    <a class="dropdown-item" href="../articles/options.html">Available Options</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ZhangLabGT/scMultiSim/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      <h2 id="simulation">Simulation<a class="anchor" aria-label="anchor" href="#simulation"></a></h2>
      
      <div class="section-desc"><p>Functions for simulating single-cell data.</p></div>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="sim_true_counts.html">sim_true_counts()</a></code> 
        </dt>
        <dd>Simulate true scRNA and scATAC counts from the parameters</dd>
      </dl><dl><dt>
          
          <code><a href="add_expr_noise.html">add_expr_noise()</a></code> 
        </dt>
        <dd>Add experimental noise to true counts</dd>
      </dl><dl><dt>
          
          <code><a href="divide_batches.html">divide_batches()</a></code> 
        </dt>
        <dd>Divide batches for observed counts</dd>
      </dl><dl><dt>
          
          <code><a href="add_outliers.html">add_outliers()</a></code> 
        </dt>
        <dd>Add outliers to the observed counts</dd>
      </dl></div><div class="section level2">
      <h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a></h2>
      
      <div class="section-desc"><p>Functions for visualizing the results.</p></div>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="plot_cell_loc.html">plot_cell_loc()</a></code> 
        </dt>
        <dd>Plot cell locations</dd>
      </dl><dl><dt>
          
          <code><a href="plot_gene_module_cor_heatmap.html">plot_gene_module_cor_heatmap()</a></code> 
        </dt>
        <dd>Plot the gene module correlation heatmap</dd>
      </dl><dl><dt>
          
          <code><a href="plot_grid.html">plot_grid()</a></code> 
        </dt>
        <dd>Plot the CCI grid</dd>
      </dl><dl><dt>
          
          <code><a href="plot_grn.html">plot_grn()</a></code> 
        </dt>
        <dd>Plot the GRN network</dd>
      </dl><dl><dt>
          
          <code><a href="plot_phyla.html">plot_phyla()</a></code> 
        </dt>
        <dd>Plot a R phylogenic tree</dd>
      </dl><dl><dt>
          
          <code><a href="plot_rna_velocity.html">plot_rna_velocity()</a></code> 
        </dt>
        <dd>Plot RNA velocity as arrows on tSNE plot</dd>
      </dl><dl><dt>
          
          <code><a href="plot_tsne.html">plot_tsne()</a></code> 
        </dt>
        <dd>Plot t-SNE visualization of a data matrix</dd>
      </dl><dl><dt>
          
          <code><a href="gene_corr_cci.html">gene_corr_cci()</a></code> 
        </dt>
        <dd>Plot the ligand-receptor correlation summary</dd>
      </dl><dl><dt>
          
          <code><a href="gene_corr_regulator.html">gene_corr_regulator()</a></code> 
        </dt>
        <dd>Print the correlations between targets of each regulator</dd>
      </dl></div><div class="section level2">
      <h2 id="help">Help<a class="anchor" aria-label="anchor" href="#help"></a></h2>
      
      <div class="section-desc"><p>Functions for getting help.</p></div>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="run_shiny.html">run_shiny()</a></code> 
        </dt>
        <dd>Launch the Shiny App to configure the simulation</dd>
      </dl><dl><dt>
          
          <code><a href="scmultisim_help.html">scmultisim_help()</a></code> 
        </dt>
        <dd>Show detailed documentations of scMultiSim's parameters</dd>
      </dl></div><div class="section level2">
      <h2 id="utilities">Utilities<a class="anchor" aria-label="anchor" href="#utilities"></a></h2>
      
      <div class="section-desc"><p>Utility functions that can be useful for simulating data.</p></div>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="cci_cell_type_params.html">cci_cell_type_params()</a></code> 
        </dt>
        <dd>Generate cell-type level CCI parameters</dd>
      </dl><dl><dt>
          
          <code><a href="gen_clutter.html">gen_clutter()</a></code> 
        </dt>
        <dd>generate a clutter of cells by growing from the center</dd>
      </dl></div><div class="section level2">
      <h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a></h2>
      
      <div class="section-desc"><p>Default data provided by scMultiSim</p></div>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="Phyla1.html">Phyla1()</a></code> 
        </dt>
        <dd>Creating a linear example tree</dd>
      </dl><dl><dt>
          
          <code><a href="Phyla3.html">Phyla3()</a></code> 
        </dt>
        <dd>Creating an example tree with 3 tips</dd>
      </dl><dl><dt>
          
          <code><a href="Phyla5.html">Phyla5()</a></code> 
        </dt>
        <dd>Creating an example tree with 5 tips</dd>
      </dl><dl><dt>
          
          <code><a href="GRN_params_100.html">GRN_params_100</a></code> 
        </dt>
        <dd>100_gene_GRN is a matrix of GRN params consisting of 100 genes where: # - column 1 is the target gene ID, # - column 2 is the gene ID which acts as a transcription factor for the target (regulated) gene # - column 3 is the effect of the column 2 gene ID on the column 1 gene ID</dd>
      </dl><dl><dt>
          
          <code><a href="GRN_params_1139.html">GRN_params_1139</a></code> 
        </dt>
        <dd>GRN_params_1139 is a matrix of GRN params consisting of 1139 genes where: # - column 1 is the target gene ID, # - column 2 is the gene ID which acts as a transcription factor for the target (regulated) gene # - column 3 is the effect of the column 2 gene ID on the column 1 gene ID</dd>
      </dl><dl><dt>
          
          <code><a href="dens_nonzero.html">dens_nonzero</a></code> 
        </dt>
        <dd>this is the density function of log(x+1), where x is the non-zero values for ATAC-SEQ data</dd>
      </dl><dl><dt>
          
          <code><a href="gene_len_pool.html">gene_len_pool</a></code> 
        </dt>
        <dd>a pool of gene lengths to sample from</dd>
      </dl><dl><dt>
          
          <code><a href="len2nfrag.html">len2nfrag</a></code> 
        </dt>
        <dd>from transcript length to number of fragments (for the nonUMI protocol)</dd>
      </dl><dl><dt>
          
          <code><a href="match_params.html">param_realdata.zeisel.imputed</a></code> 
        </dt>
        <dd>distribution of kinetic parameters learned from the Zeisel UMI cortex datasets</dd>
      </dl></div><div class="section level2">
      <h2 id="internal-helpers">Internal helpers<a class="anchor" aria-label="anchor" href="#internal-helpers"></a></h2>
      
      <div class="section-desc"><p>Internal helper functions, but can be useful for advanced customization.</p></div>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="Get_1region_ATAC_correlation.html">Get_1region_ATAC_correlation()</a></code> 
        </dt>
        <dd>This function gets the average correlation rna seq counts and region effect on genes for genes which are only associated with 1 chromatin region</dd>
      </dl><dl><dt>
          
          <code><a href="Get_ATAC_correlation.html">Get_ATAC_correlation()</a></code> 
        </dt>
        <dd>This function gets the average correlation rna seq counts and chromatin region effect on genes</dd>
      </dl><dl><dt>
          
          <code><a href="True2ObservedATAC.html">True2ObservedATAC()</a></code> 
        </dt>
        <dd>Simulate observed ATAC-seq matrix given technical noise and the true counts</dd>
      </dl><dl><dt>
          
          <code><a href="True2ObservedCounts.html">True2ObservedCounts()</a></code> 
        </dt>
        <dd>Simulate observed count matrix given technical biases and the true counts</dd>
      </dl><dl><dt>
          
          <code><a href="sim_example_200_cells.html">sim_example_200_cells()</a></code> 
        </dt>
        <dd>Simulate a small example dataset with 200 cells and the 100-gene GRN</dd>
      </dl><dl><dt>
          
          <code><a href="sim_example_200_cells_spatial.html">sim_example_200_cells_spatial()</a></code> 
        </dt>
        <dd>Simulate a small example dataset with 200 cells and the 100-gene GRN, with CCI enabled</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hechen Li, Xiuwei Zhang, Michael Squires.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

