<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Show detailed documentations of scMultiSim's parameters"><title>Show detailed documentations of scMultiSim's parameters — scmultisim_help • scMultiSim</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Show detailed documentations of scMultiSim's parameters — scmultisim_help"><meta property="og:description" content="Show detailed documentations of scMultiSim's parameters"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scMultiSim</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/workflow.html">Basic Workflow</a>
    <a class="dropdown-item" href="../articles/basics.html">Simulating Multimodal Single-cell Datasets</a>
    <a class="dropdown-item" href="../articles/spatialCCI.html">Simulating Spatial Cell-Cell Interactions</a>
    <a class="dropdown-item" href="../articles/options.html">Available Options</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ZhangLabGT/scMultiSim/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Show detailed documentations of scMultiSim's parameters</h1>
      <small class="dont-index">Source: <a href="https://github.com/ZhangLabGT/scMultiSim/blob/HEAD/R/9_meta.R" class="external-link"><code>R/9_meta.R</code></a></small>
      <div class="d-none name"><code>scmultisim_help.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Show detailed documentations of scMultiSim's parameters</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">scmultisim_help</span><span class="op">(</span>topic <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>topic</dt>
<dd><p>Can be <code>options</code>, <code>dynamic.GRN</code>, or <code>cci</code></p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>none</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">scmultisim_help</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Call scmultisim_help(topic) where topic can be "options" or an option na</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	me. Printing help for options by default.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> scMultiSim  v1.0.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [GENERAL]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rand.seed  (default: 0)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	scMultiSim should produce the same result if all other parameters are th</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	e same.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> threads  (default: 1)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Set to larger than 1 to use multithreading for some part of the simulati</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	on.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be an integer between 1 and 4096.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> speed.up  (default: FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Use experimental speed and memory optimization.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a logical.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [GENE]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRN  (default: NULL)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The GRN network.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	It should be a data frame with 3 columns (target, regulator, effect). Su</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	pply NA to disable the GRN effect.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> grn.effect  (default: 1)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Overall strength of the GRN effect on the expression. Different from the</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	 effect column in the GRN data frame, which is the relative effect of ea</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	ch TF-target pair.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 0 and Inf.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> num.genes  (default: NULL)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Number of genes if GRN is disabled.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be an integer between 1 and Inf.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> unregulated.gene.ratio  (default: 0.1)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Ratio of unreulated to regulated genes. Extra unregulated genes will be </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	simulated in addition to the genes in GRN.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 0 and 1.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> giv.mean  (default: 0)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Mean of the Gene Identity Vectors.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between -Inf and Inf.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> giv.prob  (default: 0.3)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Probability of non-zero values in the Gene Identity Vectors.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 0 and 1.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> giv.sd  (default: 1)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Stddev of the Gene Identity Vectors.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 0 and Inf.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> hge.range  (default: 1)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Only choose highly expressed genes after this range.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 1 and Inf.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> hge.prop  (default: 0)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Propotion of highly expressed genes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 0 and 1.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> hge.mean  (default: 5)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Scale of highly expressed genes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 1 and Inf.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> hge.sd  (default: 1)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Variation of highly expressed genes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 0 and Inf.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> hge.max.var  (default: 500)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Genes with higher variation will not be selected as highly expressed gen</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	es.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 0 and Inf.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dynamic.GRN  (default: NA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Specification of the dynamic GRN. See scmultisim_help("dynamic.GRN") for</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	 details.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [CELL]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> num.cells  (default: 1000)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Total number of cells from all populations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be an integer between 0 and Inf.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tree  (default: Phyla5())</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	A tree defining relationship between populations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> discrete.cif  (default: FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Whether the cell population is discrete.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a logical.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> discrete.pop.size  (default: NA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Specify the cell numbers in each population.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	the value should be an integer vector</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> discrete.min.pop.size  (default: 70)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Size of the smallest discrete cell population.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> discrete.min.pop.index  (default: 1)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Index of the smallest discrete cell population.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be an integer between 0 and Inf.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> num.cifs  (default: 50)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Number of Cell Identity Factors for each kinetic parameter.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> diff.cif.fraction  (default: 0.9)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Fraction of CIFs which are differential factors between cell types.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 0 and 1.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> cif.center  (default: 1)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Mean of the CIF values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> cif.sigma  (default: 0.1)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Stddev of the CIF values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 0 and Inf.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> use.impulse  (default: FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Use the impulse model when generating the continuous CIF.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a logical.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [SIMULATION - ATAC]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> atac.effect  (default: 0.5)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The influence of chromatin accessability data on gene expression.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 0 and 1.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> region.distrib  (default: c(0.1, 0.5, 0.4))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The probability that a gene is regulated by respectively 0, 1, 2 consecu</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	tive regions.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	the value should be a vector with 3 elements sum to 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> atac.p_zero  (default: 0.8)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The proportion of 0s we see in the ATAC-seq data.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> atac.density  (default: NA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Density of the non-zero ATAC-seq values. Use atac_dens_nonzero() to gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	rate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	the value should be a density object.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> riv.mean  (default: 0)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Mean of the Region Identity Vectors.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 0 and Inf.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> riv.prob  (default: 0.3)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Probability of non-zero values in the Region Identity Vectors.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 0 and 1.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> riv.sd  (default: 1)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Stddev of the Region Identity Vectors.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 0 and Inf.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [SIMULATION - RNA]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> vary  (default: s)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Which kinetic parameters have differential CIFs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be one of [all, kon, koff, s, except_kon, except_koff, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	except_s].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> bimod  (default: 0)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Adjust the bimodality of gene expression, thus controlling intrinsic var</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	iation.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 0 and 1.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> scale.s  (default: 1)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Scale of the s parameter. Use smaller value for cell types known to be s</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	mall (like naive cells). When discrete.cif = T, it can be a vector speci</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	fying the scale.s for each cluster.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> intrinsic.noise  (default: 1)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The weight assigned to the random sample from the Beta-Poisson distribut</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	ion, where the weight of the Beta-Poisson mean value is given a weight o</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	f (1 - intrinsic.noise).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 0 and 1.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [SIMULATION - KINETIC MODEL]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> do.velocity  (default: FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Simulate using the whole kinetic model and generate RNA velocity data.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a logical.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> beta  (default: 0.4)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Splicing rate of each gene in the kinetic model.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> d  (default: 1)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Degradation rate of each gene in the kinetic model.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> num.cycles  (default: 3)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	For velocity mode, the number of cycles run before sampling the gene exp</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	ression of a cell.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be an integer between 1 and Inf.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> cycle.len  (default: 1)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	For velocity mode, a factor multiplied by the expected time to transitio</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	n from kon to koff and back to form the the length of a cycle.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The value should be a numeric between 0 and Inf.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mod.cif.giv  (default: NA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Modify the generated CIF and GIV. The function takes four arguments: the</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	 kinetic parameter index (1=kon, 2=koff, 3=s), the current CIF matrix, t</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	he GIV matrix, and the cell metadata dataframe. It should return a list </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	of two elements: the modified CIF matrix and the modified GIV matrix.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	should be a function</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ext.cif.giv  (default: NA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Add customized CIF and GIV. The function takes one argument, the kinetic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	 parameter index (1=kon, 2=koff, 3=s). It should return a list of two el</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	ements: the extra CIF matrix (n_extra_cif x n_cells) and the GIV matrix </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	(n_genes x n_extra_cif). Return NULL for no extra CIF and GIV.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	should be a function</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [SIMULATION - SPATIAL]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> cci  (default: NA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	The regulation network for spatial cell-cell interaction.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Enables cell-cell interaction. See scmultisim_help("cci") for details.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NULL</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hechen Li, Xiuwei Zhang, Michael Squires.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

