---
title: "Available Options"
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

This article introduces the available options in `scMultiSim`.

## Options: General

###  rand.seed

> integer (default: `0`)

scMultiSim should produce the same result if all other parameters are the same.

###  threads

> integer (default: `1`)

Use multithreading only when generating the CIF matrix.
It will not speed up the simulation a lot, thus not recommended.

## Options: Genes

###  GRN

> A data frame with 3 columns as below.
> Supply `NA` to disable the GRN effect. (required)

| Column | Value                                      |
| ------ | ------------------------------------------ |
| 1      | target gene ID: `integer or character`;    |
| 2      | regulator gene ID: `integer or character`; |
| 3      | effect: `number`.                          |

If `num.genes` presents, the gene IDs should not exceed this number.
The gene IDs should start from 1 and should not ship any intermidiate numbers.

Two sample datasets `GRN_params_100` and `GRN_params_1000` from
[Dibaeinia, P., &amp; Sinha, S. (2020)](https://doi.org/10.1016/j.cels.2020.08.003) are provided for testing and inspection.

###  num.genes

> integer (default: `NULL`)

If a GRN is supplied, override the total number of genes.
It should be larger than the largest gene ID in the GRN.
Otherwise, the number of genes will be determined by `N_genes * (1 + r_u)`,
where `r_u` is `unregulated.gene.ratio`.

If GRN is disabled,
this option specifies the total number of genes.

###  unregulated.gene.ratio

> number > 0 (default: `0.1`)

Ratio of unreulated to regulated genes.
When a GRN is supplied with `N` genes,
scMultiSim will simulate `N * r_u` extra (unregulated) genes.

###  giv.mean, giv.sd, giv.prob

> (default: `0, 1, 0.3`)

The parameters used to sample the GIV matrix.
With probability `giv.prob`, the value is sampled from N(`giv.mean`, `giv.sd`).
Otherwise the value is 0.

###  dynamic.GRN

> list (default: `NULL`)

Enables dynamic (cell-specific GRN).
Run `scmultisim_help("dynamic.GRN")` to see more explaination.

###  hge.prop, hge.mean, hge.sd

> (default: `0, 5, 1`)

Treat some random genes as highly-expressed (house-keeping) genes.
A proportion of `hge.prop` genes will have expression scaled by a
multiplier sampled from N(`hge.mean`, `hge.sd`).

###  hge.range

> integer (default: `1`)

When selecting highly-expressed genes, only choose genes with ID > `hge.range`.

###  hge.max.var

> number (default: `500`)

When selecting highly-expressed genes, only choose genes
with variation < `hge.max.var`.

## Options: Cells

###  num.cells

> integer (default: `1000`)

The number of cells to be simulated.

###  tree

> phylo (default: `Phyla5()`)

The cell differential tree,
which will be used to generate cell trajectories (if `discrete.cif = T`)
or clusters (if `discrete.cif = F`).
In discrete population mode, only the tree tips will be used.
Three demo trees, `Phyla5()`, `Phyla3()` and `Phyla1()`, are provided.

###  discrete.cif

> logical (default: `FALSE`)

Whether the cell population is discrete (continuous otherwise).

###  discrete.min.pop.size, discrete.min.pop.index

> integer, integer (default: `70, 1`)

In discrete population mode, specify one cluster to have the
smallest cell population.
The cluster will contain `discrete.min.pop.size` cells.
`discrete.min.pop.index` should be a valid cluster index (tree tip number).

###  discrete.pop.size

> integer vector (default: `NA`); e.g. `c(200, 250, 300)`

Manually specify the size of each cluster.

## Options: CIF

###  num.cifs

> integer (default: `50`)

Total number of differential and non-differential CIFs,
which can be viewed as latent representation of cells.

###  diff.cif.fraction

> number (default: `0.9`)

Fraction of differential CIFs.
Differential CIFs encode the cell type information,
while non-differential CIFs are randomly sampled for each cell.

###  cif.center, cif.sigma

> (default: `1, 0.1`)

The distribution used to sample CIF values.

###  use.impulse

> logical (default: `FALSE`)

In continuous population mode, when sampling CIFs along the tree,
use the impulse model rather than the default gaussian random walk.

## Options: Simulation - ATAC

###  atac.effect

> number ∈ [0, 1] (default: `0.5`)

The influence of chromatin accessability data on gene expression.

###  region.distrib

> vector of length 3, should sum to 1 (default: `c(0.1, 0.5, 0.4)`)

The probability that a gene is regulated by 0, 1, 2
consecutive regions, respectively.

###  atac.p_zero

> number ∈ [0, 1] (default: `0.8`)

The proportion of zeros we see in the simulated scATAC-seq data.

###  riv.mean, riv.sd, riv.prob

> (default: `0, 1, 0.3`)

The parameters used to sample the RIV (Region Identity Vectors).
With probability `riv.prob`, the value is sampled from N(`riv.mean`, `riv.sd`).
Otherwise the value is 0.

## Options: Simulation - RNA

###  do.velocity

> logical (default: `FALSE`)

When set to `TRUE`,
simulate using the full kinetic model and generate RNA velocity data.
Otherwise, the Beta-Poission model will be used.

###  beta

> number (default: `0.4`)

The splicing rate of each gene in the kinetic model.

###  d

> number (default: `1`)

The degradation rate of each gene in the kinetic model.

###  num.cycles

> number (default: `3`)

The number of cycles run before sampling the gene expression of a cell.

###  cycle.len

> number (default: `1`)

In velocity mode, a multiplier for the cell cycle length.
It is multiplied by the expected time to
transition from k_on to k_off and back to form the the length of a cycle.

## Options: Simulation - Spatial Cell-Cell Interaction

###  cci

Enables cell-cell interaction. See `scmultisim_help("cci")` for details.
