<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="scMultiSim">
<title>Simulating Multimodal Single-cell Datasets • scMultiSim</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Simulating Multimodal Single-cell Datasets">
<meta property="og:description" content="scMultiSim">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scMultiSim</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/workflow.html">Getting Started</a>
    <a class="dropdown-item" href="../articles/basics.html">Simulating Multimodal Single-cell Datasets</a>
    <a class="dropdown-item" href="../articles/spatialCCI.html">Simulating Spatial Cell-Cell Interactions</a>
    <a class="dropdown-item" href="../articles/options.html">Parameter Guide</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ZhangLabGT/scMultiSim/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Simulating Multimodal Single-cell Datasets</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ZhangLabGT/scMultiSim/blob/HEAD/vignettes/basics.Rmd" class="external-link"><code>vignettes/basics.Rmd</code></a></small>
      <div class="d-none name"><code>basics.Rmd</code></div>
    </div>

    
    
<p>In this tutorial, we will demonstrate how to use scMultiSim to
simulate multi-omics data with different biological effects,
including:</p>
<ul>
<li>Simulating true RNA counts and ATAC-seq data</li>
<li>Controlling the cell population and GRN effects</li>
<li>Adding technical variation and batch effect to the true counts</li>
<li>Adjusting the parameters to control different biological
effects</li>
</ul>
<p>We first load the package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"scMultiSim"</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="simulating-true-counts">Simulating True Counts<a class="anchor" aria-label="anchor" href="#simulating-true-counts"></a>
</h2>
<p>scMultiSim first generates the true RNA counts, and then add
technical variation and batch effect to the true counts. To simulate
true counts, call <code>sim_true_counts(options)</code> where
<code>options</code> is a list. You can use
<code><a href="../reference/scmultisim_help.html">scmultisim_help()</a></code> to get help on the options, or like
<code>scmulti_help("num.cells")</code> to get help on the options for a
specific function.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/scmultisim_help.html">scmultisim_help</a></span><span class="op">(</span><span class="st">"options"</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="grn-and-differentiation-tree">GRN and Differentiation Tree<a class="anchor" aria-label="anchor" href="#grn-and-differentiation-tree"></a>
</h3>
<p>Before start, we define a utility function to modify a list.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">list_modify</span> <span class="op">&lt;-</span> <span class="kw">function</span> <span class="op">(</span><span class="va">curr_list</span>, <span class="va">...</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">args</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">...</span><span class="op">)</span></span>
<span>  <span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">args</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">curr_list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">args</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="va">curr_list</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>The minimal input to scMultiSim is a <strong>differentiation
tree</strong>, and you can optionally provide ground truth for GRN and
cell-cell interactions. The differentiation tree is an R phylo object,
which can be created using e.g. <code><a href="https://rdrr.io/pkg/ape/man/read.tree.html" class="external-link">ape::read.tree()</a></code> or
<code><a href="https://rdrr.io/pkg/ape/man/rtree.html" class="external-link">ape::rtree()</a></code>. It controls the cell population structure:
each node of the tree should represent a cell type, and connected nodes
indicate the differentiation relationship between cell types.
<em>scMultiSim provides this explicit control on the cell population
structure while preserving all other effects (such as GRN and Cell-Cell
Interactions)</em>, so you can generate any cell trajectory or
clustering structure you want, which is especially useful for
benchmarking trajectory inference and clustering methods.</p>
<p>If generating a continuous population, this tree specifies the cell
differentiation trajectory; if generating a discrete population, the
tips of this tree will be the clusters (cell types are the terminal cell
states).</p>
<p>scMultiSim also provides three differentiation trees.
<code><a href="../reference/Phyla5.html">Phyla5()</a></code> and <code><a href="../reference/Phyla3.html">Phyla3()</a></code> return bifurcating trees
with 5 and 3 leaves respectively. <code><a href="../reference/Phyla1.html">Phyla1()</a></code> returns only a
single branch, which can be useful when we don’t want any specific
trajectory.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/Phyla5.html">Phyla5</a></span><span class="op">(</span>plotting <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Phylogenetic tree with 5 tips and 4 internal nodes.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Tip labels:</span></span>
<span><span class="co">##   1, 2, 3, 4, 5</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Rooted; includes branch lengths.</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/Phyla3.html">Phyla3</a></span><span class="op">(</span>plotting <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="basics_files/figure-html/plot-tree-1.png" width="768"></p>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Phylogenetic tree with 3 tips and 2 internal nodes.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Tip labels:</span></span>
<span><span class="co">##   1, 2, 3</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Rooted; includes branch lengths.</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># It's not possible to plot Phyla1() because it only contains 1 branch connecting two nodes.</span></span>
<span><span class="fu"><a href="../reference/Phyla1.html">Phyla1</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Phylogenetic tree with 1 tips and 1 internal nodes.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Tip labels:</span></span>
<span><span class="co">##   A</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Rooted; includes branch lengths.</span></span></code></pre>
<p>If you only need <code>n</code> cell clusters without any specific
trajectory, you can use code like below to generate a simple tree with
<code>n</code> leaves.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># tree with four leaves</span></span>
<span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html" class="external-link">read.tree</a></span><span class="op">(</span>text <span class="op">=</span> <span class="st">"(A:1,B:1,C:1,D:1);"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Phylogenetic tree with 4 tips and 1 internal nodes.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Tip labels:</span></span>
<span><span class="co">##   A, B, C, D</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Unrooted; includes branch lengths.</span></span></code></pre>
<p>The GRN should be a data frame with 3 columns, each representing the
<code>target</code>, <code>regulator</code>, and <code>effect</code>.
The target and regulator should be gene names, which can be integers or
strings. The effect should be a numeric value, indicating the effect of
the regulator on the target.</p>
<p>scMultiSim provides two sample GRNs, <code>GRN_params_100</code> and
<code>GRN_params_1139</code>, which contain 100 and 1139 genes
respectively. Let’s load them first.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GRN_params_100</span><span class="op">)</span></span>
<span><span class="va">GRN_params</span> <span class="op">&lt;-</span> <span class="va">GRN_params_100</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">GRN_params</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   regulated.gene regulator.gene regulator.effect</span></span>
<span><span class="co">## 1             16             10         1.768699</span></span>
<span><span class="co">## 2             29             10         1.675360</span></span>
<span><span class="co">## 3             62             10         3.723509</span></span>
<span><span class="co">## 4             55             10         2.866182</span></span>
<span><span class="co">## 5             81             80         4.020744</span></span>
<span><span class="co">## 6             73             80         2.353371</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="simulating-true-counts-1">Simulating True Counts<a class="anchor" aria-label="anchor" href="#simulating-true-counts-1"></a>
</h3>
<p>Now, we create the options list for the simulation session. In the
following example, we simulate 500 cells with 50 CIFs.</p>
<p>The number of genes is determined by the option
<code>num.genes</code> or the number of genes in the GRN. If
<code>num.genes</code> is not specified, the number of genes will be the
number of unique genes in the GRN, plus a fraction of genes that are not
regulated by any other genes. this is controlled by the option
<code>unregulated.gene.ratio</code> (default is 0.1). Since our
<code>GRN_params</code> contains 100 gene names, 10% more genes will be
added to the simulation, and the number of genes in the simulated data
will be 110. If you don’t need to simulate GRN effects, simply set
<code>GRN = NA</code>.</p>
<p>The <code>cif.sigma</code> controls the variance of the CIFs.
Usually, with <code>cif.sigma</code> = 0.1, the trajectory will be very
clear, while with <code>cif.sigma</code> = 1, the trajectory will be
more noisy. We use <code>cif.sigma</code> = 0.5 in this example.</p>
<p>We also have <code>do.velocity</code> option to use the Kinetic model
to simulate RNA velocity data.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">options</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  rand.seed <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  GRN <span class="op">=</span> <span class="va">GRN_params</span>,</span>
<span>  num.cells <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  num.cifs <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  cif.sigma <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  tree <span class="op">=</span> <span class="fu"><a href="../reference/Phyla5.html">Phyla5</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  diff.cif.fraction <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  do.velocity <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="omitting-the-grn">Omitting the GRN<a class="anchor" aria-label="anchor" href="#omitting-the-grn"></a>
</h4>
<p>Note that the minimal input to scMultiSim is the cell population
structure (differentiation tree) and number of cells. You can omit the
GRN by using <code>GRN = NA</code>:</p>
<pre><code>options &lt;- list(
  GRN = NA
  num.cells = 1000,
  num.genes = 500,
  tree = Phyla5(),
)</code></pre>
</div>
<div class="section level4">
<h4 id="running-the-simulation">Running the Simulation<a class="anchor" aria-label="anchor" href="#running-the-simulation"></a>
</h4>
<p>Now we run the simulation and check what kind of data is in the
returned result:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sim_true_counts.html">sim_true_counts</a></span><span class="op">(</span><span class="va">options</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Time spent: 0.29 mins</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] ".grn"             "unspliced_counts" ".options"         ".n"              </span></span>
<span><span class="co">##  [5] "region_to_gene"   "atacseq_data"     "giv"              "kinetic_params"  </span></span>
<span><span class="co">##  [9] "num_genes"        "velocity"         "cif"              "hge_scale"       </span></span>
<span><span class="co">## [13] "counts"           "cell_meta"        "atac_counts"      "region_to_tf"    </span></span>
<span><span class="co">## [17] "grn_params"       "cell_time"</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="accessing-the-results">Accessing the Results<a class="anchor" aria-label="anchor" href="#accessing-the-results"></a>
</h3>
<p>The return value will be a <code>scMultiSim Environment</code>
object, and you can access various data and parameters using the
<code>$</code> operator.</p>
<ul>
<li>
<code>counts</code>: Gene-by-cell scRNA-seq counts.</li>
<li>
<code>atac_counts</code>: Region-by-cell scATAC-seq counts.</li>
<li>
<code>region_to_gene</code>: Region-by-gene 0-1 marix indicating the
corresponding relationship between chtomatin regions and genes.</li>
<li>
<code>atacseq_data</code>: The “clean” scATAC-seq counts without
added intrinsic noise.</li>
<li>
<code>cell_meta</code>: A dataframe containing cell type labels and
pseudotime information.</li>
<li>
<code>cif</code>: The CIF used during the simulation.</li>
<li>
<code>giv</code>: The GIV used during the simulation.</li>
<li>
<code>kinetic_params</code>: The kinetic parameters used during the
simulation.</li>
<li>
<code>.grn</code>: The GRN used during the simulation.</li>
<li>
<code>.grn$regulators</code>: The list of TFs used by all gene-by-TF
matrices.</li>
<li>
<code>.grn$geff</code>: Gene-by-TF matrix representing the GRN used
during the simulation.</li>
<li>
<code>.n</code>: Other metadata, e.g. <code>.n$cells</code> is the
number of cells.</li>
</ul>
<p>If <code>do.velocity</code> is enabled, it has these additional
fields:</p>
<ul>
<li>
<code>unspliced_counts</code>: Gene-by-cell unspliced RNA
counts.</li>
<li>
<code>velocity</code>: Gene-by-cell RNA velocity ground truth.</li>
<li>
<code>cell_time</code>: The pseudotime at which the cell counts were
generated.</li>
</ul>
<p>If dynamic GRN is enabled, it has these additional fields:</p>
<ul>
<li>
<code>cell_specific_grn</code>: A list of length
<code>n_cells</code>. Each element is a gene-by-TF matrix, indicating
the cell’s GRN.</li>
</ul>
<p>If cell-cell interaction is enabled, it has these additional
fields:</p>
<ul>
<li>
<code>grid</code>: The grid object used during the simulation.</li>
<li>
<code>grid$get_neighbours(i)</code>: Get the neighbour cells of cell
<code>i</code>.</li>
<li>
<code>cci_locs</code>: A dataframe containing the X and Y
coordinates of each cell.</li>
<li>
<code>cci_cell_type_param</code>: A dataframe containing the CCI
network ground truth: all ligand-receptor pairs between each pair of
cell types.</li>
<li>
<code>cci_cell_types</code>: For continuous cell population, the
sub-divided cell types along the trajectory used when simulating
CCI.</li>
</ul>
<p>If it is a debug session (<code>debug = TRUE</code>), a
<code>sim</code> field is available, which is an environment contains
all internal states and data structures.</p>
</div>
<div class="section level3">
<h3 id="visualizing-the-results">Visualizing the Results<a class="anchor" aria-label="anchor" href="#visualizing-the-results"></a>
</h3>
<p>We can visualize the true counts and ATAC-seq data using
<code><a href="../reference/plot_tsne.html">plot_tsne()</a></code>:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_tsne.html">plot_tsne</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">counts</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>         <span class="va">results</span><span class="op">$</span><span class="va">cell_meta</span><span class="op">$</span><span class="va">pop</span>,</span>
<span>         legend <span class="op">=</span> <span class="st">'pop'</span>, plot.name <span class="op">=</span> <span class="st">'True RNA Counts Tsne'</span><span class="op">)</span></span></code></pre></div>
<p><img src="basics_files/figure-html/plot-counts-1.png" width="60%"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_tsne.html">plot_tsne</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">atacseq_data</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>         <span class="va">results</span><span class="op">$</span><span class="va">cell_meta</span><span class="op">$</span><span class="va">pop</span>,</span>
<span>         legend <span class="op">=</span> <span class="st">'pop'</span>, plot.name <span class="op">=</span> <span class="st">'True ATAC-seq Tsne'</span><span class="op">)</span></span></code></pre></div>
<p><img src="basics_files/figure-html/plot-counts-2.png" width="60%"></p>
<p>Since we also have RNA velocity enabled, the <code>results</code>
also contains the following data: - <code>velocity</code>: the true RNA
velocity (genes x cells) - <code>unspliced_counts</code>: the true
unspliced RNA counts (genes x cells)</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_rna_velocity.html">plot_rna_velocity</a></span><span class="op">(</span><span class="va">results</span>, arrow.length <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $raw</span></span></code></pre>
<p><img src="basics_files/figure-html/plot-velocity-1.png" width="60%"></p>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## $normalized</span></span></code></pre>
<p><img src="basics_files/figure-html/plot-velocity-2.png" width="60%"></p>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## $knn_normalized</span></span></code></pre>
<p><img src="basics_files/figure-html/plot-velocity-3.png" width="60%"></p>
<p>We can inspect the gene-gene correlation using
<code>plot_gene_module_cor_heatmap(results)</code>:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_gene_module_cor_heatmap.html">plot_gene_module_cor_heatmap</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span></code></pre></div>
<p><img src="basics_files/figure-html/plot-gene-correlation-1.png" width="768"></p>
<pre><code><span><span class="co">## NULL</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="adding-technical-variation-and-batch-effect">Adding Technical Variation and Batch Effect<a class="anchor" aria-label="anchor" href="#adding-technical-variation-and-batch-effect"></a>
</h2>
<p>We can also add the technical variation and batch effect to the true
counts.</p>
<div class="section level3">
<h3 id="adding-technical-noise">Adding technical noise<a class="anchor" aria-label="anchor" href="#adding-technical-noise"></a>
</h3>
<p>Simply use the <code>add_expr_noise</code> function to add technical
noise to the dataset.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/add_expr_noise.html">add_expr_noise</a></span><span class="op">(</span></span>
<span>  <span class="va">results</span>,</span>
<span>  <span class="co"># options go here</span></span>
<span>  alpha_mean <span class="op">=</span> <span class="fl">1e4</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Adding experimental noise...</span></span>
<span><span class="co">## 50..100..150..200..250..300..350..400..450..500..550..600..650..700..750..800..850..900..950..1000..Using atac_counts</span></span></code></pre>
<pre><code><span><span class="co">## Time spent: 1.02 mins</span></span></code></pre>
<p>A <code>counts_obs</code> field will be added to the
<code>results</code> object.</p>
<p>This function also accepts a list of options. See the documentation
for more details.</p>
<ul>
<li>
<code>protocol</code>: <code>"umi"</code> or <code>"nonUMI"</code>,
whether simulate the UMI protocol.</li>
<li>
<code>alpha_mean</code>, <code>alpha_sd</code>: Mean and deviation
of rate of subsampling of transcripts during capture step.</li>
<li>
<code>alpha_gene_mean</code>, <code>alpha_gene_sd</code>:
<code>alpha</code> parameters, but gene-wise.</li>
<li>
<code>depth_mean</code>, <code>depth_sd</code>: Mean and deviation
of sequencing depth.</li>
<li>
<code>gene_len</code>: A vector with lengths of all genes.</li>
<li>
<code>atac.obs.prob</code>: For each integer count of a particular
region for a particular cell, the probability the count will be
observed.</li>
<li>
<code>atac.sd.frac</code>: The fraction of ATAC-seq data value used
as the standard deviation of added normally distrubted noise.</li>
<li>
<code>randseed</code>: random seed.</li>
</ul>
</div>
<div class="section level3">
<h3 id="adding-batch-effects">Adding batch effects<a class="anchor" aria-label="anchor" href="#adding-batch-effects"></a>
</h3>
<p>Finally, use the <code>divide_batches</code> function to add batch
effects.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/divide_batches.html">divide_batches</a></span><span class="op">(</span></span>
<span>  <span class="va">results</span>,</span>
<span>  nbatch <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  effect <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Adding batch effects...</span></span></code></pre>
<p>A <code>counts_with_batches</code> field will be added to the
<code>results</code> object.</p>
<p>The available options are:</p>
<ul>
<li>
<code>nbatch</code>: Number of batches.</li>
<li>
<code>effect</code>: The batch effect size.</li>
</ul>
<p>We can visualize the result with technical noise and batches:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_tsne.html">plot_tsne</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">counts_with_batches</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>          <span class="va">results</span><span class="op">$</span><span class="va">cell_meta</span><span class="op">$</span><span class="va">pop</span>,</span>
<span>          legend <span class="op">=</span> <span class="st">'pop'</span>, plot.name <span class="op">=</span> <span class="st">'RNA Counts Tsne with Batches'</span><span class="op">)</span></span></code></pre></div>
<p><img src="basics_files/figure-html/add-expr-noise-1.png" width="60%"></p>
</div>
</div>
<div class="section level2">
<h2 id="adjusting-parameters">Adjusting Parameters<a class="anchor" aria-label="anchor" href="#adjusting-parameters"></a>
</h2>
<p>scMultiSim provides various parameters to control each type of
biological effect. Here, we describe the most important parameters and
how they affect the simulation results:</p>
<ul>
<li>
<code>num.cifs</code>, <code>diff.cif.fraction</code>
</li>
<li>
<code>cif.mean</code>, <code>cif.sigma</code>
</li>
<li><code>discrete.cif</code></li>
<li><code>intinsic.noise</code></li>
</ul>
<p>For a complete list of parameters, please check out the <a href="https://zhanglabgt.github.io/scMultiSim/articles/options">Parameter
Guide</a> page in the documentation.</p>
<div class="section level3">
<h3 id="the-shiny-app">The Shiny App<a class="anchor" aria-label="anchor" href="#the-shiny-app"></a>
</h3>
<p>scMultiSim provides a Shiny app to help you generate the options list
and visualize the effects of different parameters. It is highly
recommended to use the Shiny app to explore the available parameters.
You can run the app by calling <code><a href="../reference/run_shiny.html">run_shiny()</a></code>.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/run_shiny.html">run_shiny</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="float">
<img src="https://github.com/ZhangLabGT/scMultiSim/raw/img/img/shiny_app_sc.png" alt="Shiny App"><div class="figcaption">Shiny App</div>
</div>
</div>
<div class="section level3">
<h3 id="deciding-number-of-cifs-num-cifs">Deciding Number of CIFs: <code>num.cifs</code><a class="anchor" aria-label="anchor" href="#deciding-number-of-cifs-num-cifs"></a>
</h3>
<p>In scMultiSim, user use <code>num.cifs</code> to control the total
number of diff-CIF and non-diff-CIFs. The number of CIFs should be large
enough to represent the cell population structure and gene information.
By default, <code>num.cifs</code> is set to 50, which is a good starting
point for most cases. However, each gene’s base expression is affected
by two random diff-CIF entries, therefore if you have a large number of
genes, they may have similar expression patterns, which may not be
ideal. It is recommended to increase <code>num.cifs</code> to 50-100 if
you have more than 2000 genes. If you have a small number of genes (less
than 1000), you can also decrease <code>num.cifs</code> to 20-40.</p>
</div>
<div class="section level3">
<h3 id="discrete-cell-population-discrete-cif">Discrete Cell Population: <code>discrete.cif</code><a class="anchor" aria-label="anchor" href="#discrete-cell-population-discrete-cif"></a>
</h3>
<p>We can also simulate discrete cell population by setting
<code>discrete.cif = TRUE</code>. In this case, each tip of the tree
will be one cell type, therefore there will be 5 clusters in the
following result.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">options</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  rand.seed <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  GRN <span class="op">=</span> <span class="va">GRN_params</span>,</span>
<span>  num.cells <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  num.cifs <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  tree <span class="op">=</span> <span class="fu"><a href="../reference/Phyla5.html">Phyla5</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  diff.cif.fraction <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  discrete.cif <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sim_true_counts.html">sim_true_counts</a></span><span class="op">(</span><span class="va">options</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Time spent: 0.18 mins</span></span></code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_tsne.html">plot_tsne</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">counts</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>         <span class="va">results</span><span class="op">$</span><span class="va">cell_meta</span><span class="op">$</span><span class="va">pop</span>,</span>
<span>         legend <span class="op">=</span> <span class="st">'pop'</span>, plot.name <span class="op">=</span> <span class="st">'True RNA Counts Tsne'</span><span class="op">)</span></span></code></pre></div>
<p><img src="basics_files/figure-html/simulate-discrete-1.png" width="60%"></p>
</div>
<div class="section level3">
<h3 id="adjusting-the-effect-of-cell-population-diff-cif-fraction">Adjusting the Effect of Cell Population:
<code>diff.cif.fraction</code><a class="anchor" aria-label="anchor" href="#adjusting-the-effect-of-cell-population-diff-cif-fraction"></a>
</h3>
<p>In scMultiSim, the differentiation tree provides explicit control of
the cell population. The effect of the tree can be adjusted by the
option <code>diff.cif.fraction</code>, which controls how many CIFs are
affected by the cell population. With a larger
<code>diff.cif.fraction</code>, the effect of cell population will be
larger and you may see a clearer trajectory or well separated clusters.
With a smaller <code>diff.cif.fraction</code>, the resulting RNA counts
will be more affected by other factors, such as the GRN.</p>
<p>Now let’s visualize the trajectory with different
<code>diff.cif.fraction</code> values:</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">options</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  rand.seed <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  GRN <span class="op">=</span> <span class="va">GRN_params</span>,</span>
<span>  num.cells <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  num.cifs <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  tree <span class="op">=</span> <span class="fu"><a href="../reference/Phyla5.html">Phyla5</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  diff.cif.fraction <span class="op">=</span> <span class="fl">0.8</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sim_true_counts.html">sim_true_counts</a></span><span class="op">(</span></span>
<span>        <span class="va">options</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">list_modify</span><span class="op">(</span>diff.cif.fraction <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Time spent: 0.09 mins</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_tsne.html">plot_tsne</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">counts</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>         <span class="va">results</span><span class="op">$</span><span class="va">cell_meta</span><span class="op">$</span><span class="va">pop</span>,</span>
<span>         legend <span class="op">=</span> <span class="st">'pop'</span>, plot.name <span class="op">=</span> <span class="st">'RNA Counts (diff.cif.fraction = 0.2)'</span><span class="op">)</span></span></code></pre></div>
<p><img src="basics_files/figure-html/adjust-diff-cif-fraction-1.png" width="60%"></p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sim_true_counts.html">sim_true_counts</a></span><span class="op">(</span></span>
<span>        <span class="va">options</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">list_modify</span><span class="op">(</span>diff.cif.fraction <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Time spent: 0.09 mins</span></span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_tsne.html">plot_tsne</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">counts</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>         <span class="va">results</span><span class="op">$</span><span class="va">cell_meta</span><span class="op">$</span><span class="va">pop</span>,</span>
<span>         legend <span class="op">=</span> <span class="st">'pop'</span>, plot.name <span class="op">=</span> <span class="st">'RNA Counts (diff.cif.fraction = 0.8)'</span><span class="op">)</span></span></code></pre></div>
<p><img src="basics_files/figure-html/adjust-diff-cif-fraction-2.png" width="60%"></p>
</div>
<div class="section level3">
<h3 id="adjusting-the-inherent-cell-heterogeneity-cif-mean-and-cif-sigma">Adjusting the Inherent Cell Heterogeneity: <code>cif.mean</code> and
<code>cif.sigma</code><a class="anchor" aria-label="anchor" href="#adjusting-the-inherent-cell-heterogeneity-cif-mean-and-cif-sigma"></a>
</h3>
<p>The inherent cell heterogeneity is controlled by the non-diff-CIF,
which is sampled from a normal distribution with mean
<code>cif.mean</code> and standard deviation <code>cif.sigma</code>.
Therefore, the larger <code>cif.sigma</code> is, the larger the inherent
cell heterogeneity is.</p>
<p>Now, let’s visualize the effect of <code>cif.sigma</code>:</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">options</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  rand.seed <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  GRN <span class="op">=</span> <span class="va">GRN_params</span>,</span>
<span>  num.cells <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  num.cifs <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  tree <span class="op">=</span> <span class="fu"><a href="../reference/Phyla5.html">Phyla5</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  diff.cif.fraction <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  cif.sigma <span class="op">=</span> <span class="fl">0.5</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sim_true_counts.html">sim_true_counts</a></span><span class="op">(</span></span>
<span>        <span class="va">options</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">list_modify</span><span class="op">(</span>cif.sigma <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Time spent: 0.09 mins</span></span></code></pre>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_tsne.html">plot_tsne</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">counts</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>         <span class="va">results</span><span class="op">$</span><span class="va">cell_meta</span><span class="op">$</span><span class="va">pop</span>,</span>
<span>         legend <span class="op">=</span> <span class="st">'pop'</span>, plot.name <span class="op">=</span> <span class="st">'RNA Counts (cif.sigma = 0.1)'</span><span class="op">)</span></span></code></pre></div>
<p><img src="basics_files/figure-html/adjust-cif-sigma-1.png" width="60%"></p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sim_true_counts.html">sim_true_counts</a></span><span class="op">(</span></span>
<span>        <span class="va">options</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">list_modify</span><span class="op">(</span>cif.sigma <span class="op">=</span> <span class="fl">1.0</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Time spent: 0.09 mins</span></span></code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_tsne.html">plot_tsne</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">counts</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>         <span class="va">results</span><span class="op">$</span><span class="va">cell_meta</span><span class="op">$</span><span class="va">pop</span>,</span>
<span>         legend <span class="op">=</span> <span class="st">'pop'</span>, plot.name <span class="op">=</span> <span class="st">'RNA Counts (cif.sigma = 1.0)'</span><span class="op">)</span></span></code></pre></div>
<p><img src="basics_files/figure-html/adjust-cif-sigma-2.png" width="60%"></p>
</div>
<div class="section level3">
<h3 id="adjusting-the-intrinsic-noise-intinsic-noise">Adjusting the Intrinsic Noise: <code>intinsic.noise</code><a class="anchor" aria-label="anchor" href="#adjusting-the-intrinsic-noise-intinsic-noise"></a>
</h3>
<p>If we set <code>do.velocity = FALSE</code>, scMultiSim will simulate
the RNA counts using the Beta-Poisson model, which is faster but doesn’t
output RNA velocity. When using the Beta-Possion model, scMultiSim
provides a <code>intrinsic.noise</code> parameter to control the
intrinsic noise during the transcription process. By default,
<code>intrinsic.noise</code> is set to 1, which means the true counts
will be sampled from the Beta-Poisson model. If we set
<code>intrinsic.noise</code> to a smaller value like 0.5, the true
counts will be 0.5 * (theoretical mean) + 0.5 * (sampled from the
Beta-Poisson model). <em>More intrinsic noise will make the encoded
effects (e.g. GRN) harder to be inferred.</em></p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">options</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  rand.seed <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  GRN <span class="op">=</span> <span class="va">GRN_params</span>,</span>
<span>  num.cells <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  num.cifs <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  tree <span class="op">=</span> <span class="fu"><a href="../reference/Phyla5.html">Phyla5</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  diff.cif.fraction <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  intrinsic.noise <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sim_true_counts.html">sim_true_counts</a></span><span class="op">(</span></span>
<span>        <span class="va">options</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">list_modify</span><span class="op">(</span>intrinsic.noise <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Time spent: 0.09 mins</span></span></code></pre>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_tsne.html">plot_tsne</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">counts</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>         <span class="va">results</span><span class="op">$</span><span class="va">cell_meta</span><span class="op">$</span><span class="va">pop</span>,</span>
<span>         legend <span class="op">=</span> <span class="st">'pop'</span>, plot.name <span class="op">=</span> <span class="st">'RNA Counts (intrinsic.noise = 0.5)'</span><span class="op">)</span></span></code></pre></div>
<p><img src="basics_files/figure-html/adjust-intrinsic-noise-1.png" width="60%"></p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sim_true_counts.html">sim_true_counts</a></span><span class="op">(</span></span>
<span>        <span class="va">options</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">list_modify</span><span class="op">(</span>intrinsic.noise <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Time spent: 0.09 mins</span></span></code></pre>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_tsne.html">plot_tsne</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">counts</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>         <span class="va">results</span><span class="op">$</span><span class="va">cell_meta</span><span class="op">$</span><span class="va">pop</span>,</span>
<span>         legend <span class="op">=</span> <span class="st">'pop'</span>, plot.name <span class="op">=</span> <span class="st">'RNA Counts (intrinsic.noise = 1)'</span><span class="op">)</span></span></code></pre></div>
<p><img src="basics_files/figure-html/adjust-intrinsic-noise-2.png" width="60%"></p>
</div>
<div class="section level3">
<h3 id="adjust-the-effect-of-chromatin-accessibility-atac-effect">Adjust the effect of chromatin accessibility:
<code>atac.effect</code><a class="anchor" aria-label="anchor" href="#adjust-the-effect-of-chromatin-accessibility-atac-effect"></a>
</h3>
<p><code>atac.effect</code> Controls the contribution of the chromatin
accessibility. A higher <code>atac.effect</code> means the RNA counts
are more affected by the ATAC-seq data, therefore the correlation
between the ATAC-seq and RNA-seq data will be higher.</p>
</div>
</div>
<div class="section level2">
<h2 id="simulating-dynamic-grn">Simulating Dynamic GRN<a class="anchor" aria-label="anchor" href="#simulating-dynamic-grn"></a>
</h2>
<p>First, call the following function to check the usage of dynamic
GRN.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/scmultisim_help.html">scmultisim_help</a></span><span class="op">(</span><span class="st">"dynamic.GRN"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Dynamic GRN deletes and creates some edges in the GRN in each epoch.</span></span>
<span><span class="co">## One epoch contains multiple steps, and the change is done gradually in steps.</span></span>
<span><span class="co">## The specific GRN at each step will be used by one or more cells sequentially.</span></span>
<span><span class="co">## When an epoch is done, another epoch will start.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Available options for dynamic.GRN:</span></span>
<span><span class="co">##   - seed: the random seed</span></span>
<span><span class="co">##   - num.steps: number of steps in each epoch.</span></span>
<span><span class="co">##   - cell.per.step: how many cells share the GRN in the same step.</span></span>
<span><span class="co">##   - involved.genes: a new edge will only be created within these specified genes.</span></span>
<span><span class="co">##       The default value is NA, which will use all existing genes in the GRN.</span></span>
<span><span class="co">##   - num.changing.edges: if &lt; 1, it means the portion of edges added/deleted in each epoch.</span></span>
<span><span class="co">##       if &gt;= 1, it means the number of edges added/deleted in each epoch.</span></span>
<span><span class="co">##   - create.tf.edges: whether a new edge can connect two TFs in the GRN.</span></span>
<span><span class="co">##   - weight.mean: the mean value of the weight for a newly created edge.</span></span>
<span><span class="co">##       The default value is NA, meaning that it will use the mean value of the input GRN.</span></span>
<span><span class="co">##   - weight.sd: the standard deviation of the weight for a newly created edge.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## See the returned list for the default values.</span></span></code></pre>
<pre><code><span><span class="co">## NULL</span></span></code></pre>
<p>Here we use <code><a href="../reference/Phyla1.html">Phyla1()</a></code> as the differentiation tree to
remove the effect of the trajectory. Additionally, we can use
<code>discrete.cif = TRUE</code> to simulate discrete cell
population.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">options_</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  rand.seed <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  GRN <span class="op">=</span> <span class="va">GRN_params</span>,</span>
<span>  num.cells <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  num.cifs <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  tree <span class="op">=</span> <span class="fu"><a href="../reference/Phyla1.html">Phyla1</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  diff.cif.fraction <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  do.velocity <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  dynamic.GRN <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    cell.per.step <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    num.changing.edges <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    weight.mean <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    weight.sd <span class="op">=</span> <span class="fl">4</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sim_true_counts.html">sim_true_counts</a></span><span class="op">(</span><span class="va">options_</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Time spent: 0.09 mins</span></span></code></pre>
<p><code>results$cell_specific_grn</code> is a list containing the gene
effects matrix for each cell. Each row is a target and each column is a
regulator. The corresponding gene names are displayed as column and row
names.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># GRN for cell 1 (first 10 rows)</span></span>
<span><span class="va">results</span><span class="op">$</span><span class="va">cell_specific_grn</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##           2        6       10       19 80 91</span></span>
<span><span class="co">## 1  2.250484 1.824607 1.941613 4.268812  0  0</span></span>
<span><span class="co">## 2  0.000000 0.000000 0.000000 0.000000  0  0</span></span>
<span><span class="co">## 3  2.184071 0.000000 0.000000 0.000000  0  0</span></span>
<span><span class="co">## 4  1.634807 0.000000 0.000000 0.000000  0  0</span></span>
<span><span class="co">## 5  4.677883 4.301893 2.090473 4.047174  0  0</span></span>
<span><span class="co">## 6  0.000000 0.000000 0.000000 0.000000  0  0</span></span>
<span><span class="co">## 7  3.306285 2.673968 2.414780 3.548154  0  0</span></span>
<span><span class="co">## 8  2.707510 4.653263 1.357980 1.829304  0  0</span></span>
<span><span class="co">## 9  1.501430 1.970649 4.359262 3.915318  0  0</span></span>
<span><span class="co">## 10 0.000000 0.000000 0.000000 0.000000  0  0</span></span></code></pre>
<p>Since we set <code>cell.per.step = 3</code>, we expect each adjacent
3 cells share the same GRN:</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">cell_specific_grn</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">==</span> <span class="va">results</span><span class="op">$</span><span class="va">cell_specific_grn</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">cell_specific_grn</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="op">==</span> <span class="va">results</span><span class="op">$</span><span class="va">cell_specific_grn</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">cell_specific_grn</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span> <span class="op">==</span> <span class="va">results</span><span class="op">$</span><span class="va">cell_specific_grn</span><span class="op">[[</span><span class="fl">4</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] FALSE</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">## Platform: aarch64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">## Running under: macOS 14.6.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: America/Toronto</span></span>
<span><span class="co">## tzcode source: internal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] dplyr_1.1.4      scMultiSim_1.2.0 testthat_3.2.1   knitr_1.45      </span></span>
<span><span class="co">## [5] BiocStyle_2.30.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] mnormt_2.1.1                bitops_1.0-7               </span></span>
<span><span class="co">##   [3] remotes_2.4.2.1             phangorn_2.11.1            </span></span>
<span><span class="co">##   [5] rlang_1.1.4                 magrittr_2.0.3             </span></span>
<span><span class="co">##   [7] matrixStats_1.3.0           compiler_4.3.0             </span></span>
<span><span class="co">##   [9] systemfonts_1.0.5           vctrs_0.6.5                </span></span>
<span><span class="co">##  [11] maps_3.4.2                  combinat_0.0-8             </span></span>
<span><span class="co">##  [13] quadprog_1.5-8              stringr_1.5.1              </span></span>
<span><span class="co">##  [15] profvis_0.3.8               crayon_1.5.3               </span></span>
<span><span class="co">##  [17] pkgconfig_2.0.3             fastmap_1.1.1              </span></span>
<span><span class="co">##  [19] XVector_0.42.0              ellipsis_0.3.2             </span></span>
<span><span class="co">##  [21] labeling_0.4.3              caTools_1.18.2             </span></span>
<span><span class="co">##  [23] utf8_1.2.4                  promises_1.2.1             </span></span>
<span><span class="co">##  [25] rmarkdown_2.28              markdown_1.13              </span></span>
<span><span class="co">##  [27] sessioninfo_1.2.2           ragg_1.2.7                 </span></span>
<span><span class="co">##  [29] purrr_1.0.2                 xfun_0.45                  </span></span>
<span><span class="co">##  [31] zlibbioc_1.48.2             cachem_1.0.8               </span></span>
<span><span class="co">##  [33] clusterGeneration_1.3.8     GenomeInfoDb_1.38.8        </span></span>
<span><span class="co">##  [35] jsonlite_1.8.8              highr_0.10                 </span></span>
<span><span class="co">##  [37] later_1.3.2                 DelayedArray_0.28.0        </span></span>
<span><span class="co">##  [39] KernelKnn_1.1.5             parallel_4.3.0             </span></span>
<span><span class="co">##  [41] R6_2.5.1                    bslib_0.6.1                </span></span>
<span><span class="co">##  [43] stringi_1.8.3               pkgload_1.3.4              </span></span>
<span><span class="co">##  [45] brio_1.1.4                  GenomicRanges_1.54.1       </span></span>
<span><span class="co">##  [47] jquerylib_0.1.4             numDeriv_2016.8-1.1        </span></span>
<span><span class="co">##  [49] assertthat_0.2.1            Rcpp_1.0.12                </span></span>
<span><span class="co">##  [51] bookdown_0.37               SummarizedExperiment_1.32.0</span></span>
<span><span class="co">##  [53] iterators_1.0.14            optimParallel_1.0-2        </span></span>
<span><span class="co">##  [55] usethis_2.2.2               IRanges_2.36.0             </span></span>
<span><span class="co">##  [57] httpuv_1.6.14               Matrix_1.6-5               </span></span>
<span><span class="co">##  [59] igraph_2.0.3                tidyselect_1.2.1           </span></span>
<span><span class="co">##  [61] abind_1.4-5                 rstudioapi_0.15.0          </span></span>
<span><span class="co">##  [63] yaml_2.3.8                  gplots_3.1.3.1             </span></span>
<span><span class="co">##  [65] doParallel_1.0.17           codetools_0.2-19           </span></span>
<span><span class="co">##  [67] miniUI_0.1.1.1              pkgbuild_1.4.3             </span></span>
<span><span class="co">##  [69] lattice_0.22-5              tibble_3.2.1               </span></span>
<span><span class="co">##  [71] Biobase_2.62.0              shiny_1.8.0                </span></span>
<span><span class="co">##  [73] withr_3.0.0                 coda_0.19-4.1              </span></span>
<span><span class="co">##  [75] evaluate_0.23               Rtsne_0.17                 </span></span>
<span><span class="co">##  [77] phytools_2.3-0              desc_1.4.3                 </span></span>
<span><span class="co">##  [79] urlchecker_1.0.1            pillar_1.9.0               </span></span>
<span><span class="co">##  [81] BiocManager_1.30.22         MatrixGenerics_1.14.0      </span></span>
<span><span class="co">##  [83] KernSmooth_2.23-22          foreach_1.5.2              </span></span>
<span><span class="co">##  [85] stats4_4.3.0                generics_0.1.3             </span></span>
<span><span class="co">##  [87] rprojroot_2.0.4             RCurl_1.98-1.16            </span></span>
<span><span class="co">##  [89] S4Vectors_0.40.2            ggplot2_3.5.1              </span></span>
<span><span class="co">##  [91] munsell_0.5.1               scales_1.3.0               </span></span>
<span><span class="co">##  [93] gtools_3.9.5                xtable_1.8-4               </span></span>
<span><span class="co">##  [95] glue_1.7.0                  scatterplot3d_0.3-44       </span></span>
<span><span class="co">##  [97] tools_4.3.0                 fs_1.6.3                   </span></span>
<span><span class="co">##  [99] fastmatch_1.1-4             grid_4.3.0                 </span></span>
<span><span class="co">## [101] ape_5.8                     devtools_2.4.5             </span></span>
<span><span class="co">## [103] colorspace_2.1-0            nlme_3.1-164               </span></span>
<span><span class="co">## [105] GenomeInfoDbData_1.2.11     cli_3.6.3                  </span></span>
<span><span class="co">## [107] DEoptim_2.2-8               textshaping_0.3.7          </span></span>
<span><span class="co">## [109] fansi_1.0.6                 expm_0.999-9               </span></span>
<span><span class="co">## [111] S4Arrays_1.2.1              gtable_0.3.5               </span></span>
<span><span class="co">## [113] zeallot_0.1.0               sass_0.4.8                 </span></span>
<span><span class="co">## [115] digest_0.6.36               BiocGenerics_0.48.1        </span></span>
<span><span class="co">## [117] SparseArray_1.2.4           farver_2.1.2               </span></span>
<span><span class="co">## [119] htmlwidgets_1.6.4           memoise_2.0.1              </span></span>
<span><span class="co">## [121] htmltools_0.5.7             pkgdown_2.0.9              </span></span>
<span><span class="co">## [123] lifecycle_1.0.4             mime_0.12                  </span></span>
<span><span class="co">## [125] MASS_7.3-60.0.1</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hechen Li, Xiuwei Zhang, Ziqi Zhang, Michael Squires.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
