<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="scMultiSim">
<title>Parameter Guide • scMultiSim</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Parameter Guide">
<meta property="og:description" content="scMultiSim">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scMultiSim</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/workflow.html">Getting Started</a>
    <a class="dropdown-item" href="../articles/basics.html">Simulating Multimodal Single-cell Datasets</a>
    <a class="dropdown-item" href="../articles/spatialCCI.html">Simulating Spatial Cell-Cell Interactions</a>
    <a class="dropdown-item" href="../articles/options.html">Parameter Guide</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ZhangLabGT/scMultiSim/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Parameter Guide</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ZhangLabGT/scMultiSim/blob/HEAD/vignettes/options.Rmd" class="external-link"><code>vignettes/options.Rmd</code></a></small>
      <div class="d-none name"><code>options.Rmd</code></div>
    </div>

    
    
<p>This article introduces the available options in
<code>scMultiSim</code>.</p>
<p>The following flow chart shows the workflow of
<code>scMultiSim</code> and each parameter’s role in the simulation.</p>
<div class="float">
<img src="https://github.com/ZhangLabGT/scMultiSim/raw/img/img/params.png" alt="scMultiSim parameters flow chart"><div class="figcaption">scMultiSim parameters flow chart</div>
</div>
<div class="section level2">
<h2 id="options-general">Options: General<a class="anchor" aria-label="anchor" href="#options-general"></a>
</h2>
<div class="section level3">
<h3 id="rand-seed">rand.seed<a class="anchor" aria-label="anchor" href="#rand-seed"></a>
</h3>
<blockquote>
<p>integer (default: <code>0</code>)</p>
</blockquote>
<p>scMultiSim should produce the same result if all other parameters are
the same.</p>
</div>
<div class="section level3">
<h3 id="threads">threads<a class="anchor" aria-label="anchor" href="#threads"></a>
</h3>
<blockquote>
<p>integer (default: <code>1</code>)</p>
</blockquote>
<p>Use multithreading only when generating the CIF matrix. It will not
speed up the simulation a lot, thus not recommended.</p>
</div>
<div class="section level3">
<h3 id="speed-up">speed.up<a class="anchor" aria-label="anchor" href="#speed-up"></a>
</h3>
<blockquote>
<p>logical (default: <code>FALSE</code>)</p>
</blockquote>
<p>Enable experimental speed-up mode. It is recommended to
<strong>enable</strong> this option, and it will be the default in the
future. Currently, it is disabled for reproducibility.</p>
</div>
</div>
<div class="section level2">
<h2 id="options-genes">Options: Genes<a class="anchor" aria-label="anchor" href="#options-genes"></a>
</h2>
<div class="section level3">
<h3 id="grn">GRN<a class="anchor" aria-label="anchor" href="#grn"></a>
</h3>
<blockquote>
<p>A data frame with 3 columns as below. Supply <code>NA</code> to
disable the GRN effect. (required)</p>
</blockquote>
<table class="table">
<thead><tr class="header">
<th>Column</th>
<th>Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>target gene ID: <code>integer or character</code>;</td>
</tr>
<tr class="even">
<td>2</td>
<td>regulator gene ID: <code>integer or character</code>;</td>
</tr>
<tr class="odd">
<td>3</td>
<td>effect: <code>number</code>.</td>
</tr>
</tbody>
</table>
<p>If <code>num.genes</code> presents, the gene IDs should not exceed
this number. The gene IDs should start from 1 and should not ship any
intermidiate numbers.</p>
<p>Two sample datasets <code>GRN_params_100</code> and
<code>GRN_params_1000</code> from <a href="https://doi.org/10.1016/j.cels.2020.08.003" class="external-link">Dibaeinia, P., &amp;
Sinha, S. (2020)</a> are provided for testing and inspection.</p>
</div>
<div class="section level3">
<h3 id="num-genes">num.genes<a class="anchor" aria-label="anchor" href="#num-genes"></a>
</h3>
<blockquote>
<p>integer (default: <code>NULL</code>)</p>
</blockquote>
<p>If a GRN is supplied, override the total number of genes. It should
be larger than the largest gene ID in the GRN. Otherwise, the number of
genes will be determined by <code>N_genes * (1 + r_u)</code>, where
<code>r_u</code> is <code>unregulated.gene.ratio</code>.</p>
<p>If GRN is disabled, this option specifies the total number of
genes.</p>
</div>
<div class="section level3">
<h3 id="unregulated-gene-ratio">unregulated.gene.ratio<a class="anchor" aria-label="anchor" href="#unregulated-gene-ratio"></a>
</h3>
<blockquote>
<p>number &gt; 0 (default: <code>0.1</code>)</p>
</blockquote>
<p>Ratio of unreulated to regulated genes. When a GRN is supplied with
<code>N</code> genes, scMultiSim will simulate <code>N * r_u</code>
extra (unregulated) genes.</p>
</div>
<div class="section level3">
<h3 id="giv-mean-giv-sd-giv-prob">giv.mean, giv.sd, giv.prob<a class="anchor" aria-label="anchor" href="#giv-mean-giv-sd-giv-prob"></a>
</h3>
<blockquote>
<p>(default: <code>0, 1, 0.3</code>)</p>
</blockquote>
<p>The parameters used to sample the GIV matrix. With probability
<code>giv.prob</code>, the value is sampled from
N(<code>giv.mean</code>, <code>giv.sd</code>). Otherwise the value is
0.</p>
</div>
<div class="section level3">
<h3 id="dynamic-grn">dynamic.GRN<a class="anchor" aria-label="anchor" href="#dynamic-grn"></a>
</h3>
<blockquote>
<p>list (default: <code>NULL</code>)</p>
</blockquote>
<p>Enables dynamic (cell-specific GRN). Run
<code>scmultisim_help("dynamic.GRN")</code> to see more
explaination.</p>
</div>
<div class="section level3">
<h3 id="hge-prop-hge-mean-hge-sd">hge.prop, hge.mean, hge.sd<a class="anchor" aria-label="anchor" href="#hge-prop-hge-mean-hge-sd"></a>
</h3>
<blockquote>
<p>(default: <code>0, 5, 1</code>)</p>
</blockquote>
<p>Treat some random genes as highly-expressed (house-keeping) genes. A
proportion of <code>hge.prop</code> genes will have expression scaled by
a multiplier sampled from N(<code>hge.mean</code>,
<code>hge.sd</code>).</p>
</div>
<div class="section level3">
<h3 id="hge-range">hge.range<a class="anchor" aria-label="anchor" href="#hge-range"></a>
</h3>
<blockquote>
<p>integer (default: <code>1</code>)</p>
</blockquote>
<p>When selecting highly-expressed genes, only choose genes with ID &gt;
<code>hge.range</code>.</p>
</div>
<div class="section level3">
<h3 id="hge-max-var">hge.max.var<a class="anchor" aria-label="anchor" href="#hge-max-var"></a>
</h3>
<blockquote>
<p>number (default: <code>500</code>)</p>
</blockquote>
<p>When selecting highly-expressed genes, only choose genes with
variation &lt; <code>hge.max.var</code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="options-cells">Options: Cells<a class="anchor" aria-label="anchor" href="#options-cells"></a>
</h2>
<div class="section level3">
<h3 id="num-cells">num.cells<a class="anchor" aria-label="anchor" href="#num-cells"></a>
</h3>
<blockquote>
<p>integer (default: <code>1000</code>)</p>
</blockquote>
<p>The number of cells to be simulated.</p>
</div>
<div class="section level3">
<h3 id="tree">tree<a class="anchor" aria-label="anchor" href="#tree"></a>
</h3>
<blockquote>
<p>phylo (default: <code><a href="../reference/Phyla5.html">Phyla5()</a></code>)</p>
</blockquote>
<p>The cell differential tree, which will be used to generate cell
trajectories (if <code>discrete.cif = T</code>) or clusters (if
<code>discrete.cif = F</code>). In discrete population mode, only the
tree tips will be used. Three demo trees, <code><a href="../reference/Phyla5.html">Phyla5()</a></code>,
<code><a href="../reference/Phyla3.html">Phyla3()</a></code> and <code><a href="../reference/Phyla1.html">Phyla1()</a></code>, are provided.</p>
</div>
<div class="section level3">
<h3 id="discrete-cif">discrete.cif<a class="anchor" aria-label="anchor" href="#discrete-cif"></a>
</h3>
<blockquote>
<p>logical (default: <code>FALSE</code>)</p>
</blockquote>
<p>Whether the cell population is discrete (continuous otherwise).</p>
</div>
<div class="section level3">
<h3 id="discrete-min-pop-size-discrete-min-pop-index">discrete.min.pop.size, discrete.min.pop.index<a class="anchor" aria-label="anchor" href="#discrete-min-pop-size-discrete-min-pop-index"></a>
</h3>
<blockquote>
<p>integer, integer (default: <code>70, 1</code>)</p>
</blockquote>
<p>In discrete population mode, specify one cluster to have the smallest
cell population. The cluster will contain
<code>discrete.min.pop.size</code> cells.
<code>discrete.min.pop.index</code> should be a valid cluster index
(tree tip number).</p>
</div>
<div class="section level3">
<h3 id="discrete-pop-size">discrete.pop.size<a class="anchor" aria-label="anchor" href="#discrete-pop-size"></a>
</h3>
<blockquote>
<p>integer vector (default: <code>NA</code>);
e.g. <code>c(200, 250, 300)</code></p>
</blockquote>
<p>Manually specify the size of each cluster.</p>
</div>
</div>
<div class="section level2">
<h2 id="options-cif">Options: CIF<a class="anchor" aria-label="anchor" href="#options-cif"></a>
</h2>
<div class="section level3">
<h3 id="num-cifs">num.cifs<a class="anchor" aria-label="anchor" href="#num-cifs"></a>
</h3>
<blockquote>
<p>integer (default: <code>50</code>)</p>
</blockquote>
<p>Total number of differential and non-differential CIFs, which can be
viewed as latent representation of cells.</p>
</div>
<div class="section level3">
<h3 id="diff-cif-fraction">diff.cif.fraction<a class="anchor" aria-label="anchor" href="#diff-cif-fraction"></a>
</h3>
<blockquote>
<p>number (default: <code>0.9</code>)</p>
</blockquote>
<p>Fraction of differential CIFs. Differential CIFs encode the cell type
information, while non-differential CIFs are randomly sampled for each
cell.</p>
</div>
<div class="section level3">
<h3 id="cif-center-cif-sigma">cif.center, cif.sigma<a class="anchor" aria-label="anchor" href="#cif-center-cif-sigma"></a>
</h3>
<blockquote>
<p>(default: <code>1, 0.1</code>)</p>
</blockquote>
<p>The distribution used to sample CIF values.</p>
</div>
<div class="section level3">
<h3 id="use-impulse">use.impulse<a class="anchor" aria-label="anchor" href="#use-impulse"></a>
</h3>
<blockquote>
<p>logical (default: <code>FALSE</code>)</p>
</blockquote>
<p>In continuous population mode, when sampling CIFs along the tree, use
the impulse model rather than the default gaussian random walk.</p>
</div>
</div>
<div class="section level2">
<h2 id="options-simulation---atac">Options: Simulation - ATAC<a class="anchor" aria-label="anchor" href="#options-simulation---atac"></a>
</h2>
<div class="section level3">
<h3 id="atac-effect">atac.effect<a class="anchor" aria-label="anchor" href="#atac-effect"></a>
</h3>
<blockquote>
<p>number ∈ [0, 1] (default: <code>0.5</code>)</p>
</blockquote>
<p>The influence of chromatin accessability data on gene expression.</p>
</div>
<div class="section level3">
<h3 id="region-distrib">region.distrib<a class="anchor" aria-label="anchor" href="#region-distrib"></a>
</h3>
<blockquote>
<p>vector of length 3, should sum to 1 (default:
<code>c(0.1, 0.5, 0.4)</code>)</p>
</blockquote>
<p>The probability that a gene is regulated by 0, 1, 2 consecutive
regions, respectively.</p>
</div>
<div class="section level3">
<h3 id="atac-p_zero">atac.p_zero<a class="anchor" aria-label="anchor" href="#atac-p_zero"></a>
</h3>
<blockquote>
<p>number ∈ [0, 1] (default: <code>0.8</code>)</p>
</blockquote>
<p>The proportion of zeros we see in the simulated scATAC-seq data.</p>
</div>
<div class="section level3">
<h3 id="riv-mean-riv-sd-riv-prob">riv.mean, riv.sd, riv.prob<a class="anchor" aria-label="anchor" href="#riv-mean-riv-sd-riv-prob"></a>
</h3>
<blockquote>
<p>(default: <code>0, 1, 0.3</code>)</p>
</blockquote>
<p>The parameters used to sample the RIV (Region Identity Vectors). With
probability <code>riv.prob</code>, the value is sampled from
N(<code>riv.mean</code>, <code>riv.sd</code>). Otherwise the value is
0.</p>
</div>
</div>
<div class="section level2">
<h2 id="customization">Customization<a class="anchor" aria-label="anchor" href="#customization"></a>
</h2>
<div class="section level3">
<h3 id="mod-cif-giv">mod.cif.giv<a class="anchor" aria-label="anchor" href="#mod-cif-giv"></a>
</h3>
<blockquote>
<p>function (default: <code>NULL</code>)</p>
</blockquote>
<p>Modify the generated CIF and GIV. The function takes four arguments:
the kinetic parameter index (1=kon, 2=koff, 3=s), the current CIF
matrix, the GIV matrix, and the cell metadata dataframe. It should
return a list of two elements: the modified CIF matrix and the modified
GIV matrix.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sim_true_counts.html">sim_true_counts</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    <span class="co"># ...</span></span>
<span>    mod.cif.giv <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">i</span>, <span class="va">cif</span>, <span class="va">giv</span>, <span class="va">meta</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="co"># modify cif and giv</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">cif</span>, <span class="va">giv</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="ext-cif-giv">ext.cif.giv<a class="anchor" aria-label="anchor" href="#ext-cif-giv"></a>
</h3>
<blockquote>
<p>function (default: <code>NULL</code>)</p>
</blockquote>
<p>Add extra CIF and GIV. The function takes one argument, the kinetic
parameter index (1=kon, 2=koff, 3=s). It should return a list of two
elements: the extra CIF matrix <code>(n_extra_cif x n_cells)</code> and
the GIV matrix <code>(n_genes x n_extra_cif)</code>. Return
<code>NULL</code> for no extra CIF and GIV.”</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sim_true_counts.html">sim_true_counts</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    <span class="co"># ...</span></span>
<span>    ext.cif.giv <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="co"># add extra cif and giv</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">extra_cif</span>, <span class="va">extra_giv</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="optins-simulation">Optins: Simulation<a class="anchor" aria-label="anchor" href="#optins-simulation"></a>
</h2>
<div class="section level3">
<h3 id="vary">vary<a class="anchor" aria-label="anchor" href="#vary"></a>
</h3>
<blockquote>
<p>character (default: <code>"s"</code>)</p>
</blockquote>
<p>Can be
<code>"all", "kon", "koff", "s", "except_kon", "except_koff", "except_s"</code>.
It specifies which kinetic parameters to vary across cells, i.e. which
kinetic parameters have differential CIFs sampled from the tree.</p>
</div>
<div class="section level3">
<h3 id="bimod">bimod<a class="anchor" aria-label="anchor" href="#bimod"></a>
</h3>
<blockquote>
<p>number (default: <code>0</code>)</p>
</blockquote>
<p>A number between 0 and 1, which adjust the bimodality of the gene
expression distribution.</p>
</div>
<div class="section level3">
<h3 id="scale-s">scale.s<a class="anchor" aria-label="anchor" href="#scale-s"></a>
</h3>
<blockquote>
<p>number (default: <code>1</code>)</p>
</blockquote>
<p>Manually scale the final <code>s</code> parameter, thus the gene
expression. When discrete.cif = T, it can be a vector specifying the
scale.s for each cluster. In this case, you can use smaller value for
cell types known to be small (like naive cells).</p>
</div>
<div class="section level3">
<h3 id="intrinsic-noise">intrinsic.noise<a class="anchor" aria-label="anchor" href="#intrinsic-noise"></a>
</h3>
<blockquote>
<p>number (default: <code>1</code>)</p>
</blockquote>
<p>A number between 0 and 1, which specify the weight of the random
sample from the Beta-Poisson distribution.</p>
<pre><code>       0 &lt;----------------------&gt; 1
Theoritical mean          Random sample from
                      Beta-Poisson distribution</code></pre>
</div>
</div>
<div class="section level2">
<h2 id="options-simulation---rna-velocity">Options: Simulation - RNA Velocity<a class="anchor" aria-label="anchor" href="#options-simulation---rna-velocity"></a>
</h2>
<div class="section level3">
<h3 id="do-velocity">do.velocity<a class="anchor" aria-label="anchor" href="#do-velocity"></a>
</h3>
<blockquote>
<p>logical (default: <code>FALSE</code>)</p>
</blockquote>
<p>When set to <code>TRUE</code>, simulate using the full kinetic model
and generate RNA velocity data. Otherwise, the Beta-Poission model will
be used.</p>
</div>
<div class="section level3">
<h3 id="beta">beta<a class="anchor" aria-label="anchor" href="#beta"></a>
</h3>
<blockquote>
<p>number (default: <code>0.4</code>)</p>
</blockquote>
<p>The splicing rate of each gene in the kinetic model.</p>
</div>
<div class="section level3">
<h3 id="d">d<a class="anchor" aria-label="anchor" href="#d"></a>
</h3>
<blockquote>
<p>number (default: <code>1</code>)</p>
</blockquote>
<p>The degradation rate of each gene in the kinetic model.</p>
</div>
<div class="section level3">
<h3 id="num-cycles">num.cycles<a class="anchor" aria-label="anchor" href="#num-cycles"></a>
</h3>
<blockquote>
<p>number (default: <code>3</code>)</p>
</blockquote>
<p>The number of cycles run before sampling the gene expression of a
cell.</p>
</div>
<div class="section level3">
<h3 id="cycle-len">cycle.len<a class="anchor" aria-label="anchor" href="#cycle-len"></a>
</h3>
<blockquote>
<p>number (default: <code>1</code>)</p>
</blockquote>
<p>In velocity mode, a multiplier for the cell cycle length. It is
multiplied by the expected time to transition from k_on to k_off and
back to form the the length of a cycle.</p>
</div>
</div>
<div class="section level2">
<h2 id="options-simulation---spatial-cell-cell-interaction">Options: Simulation - Spatial Cell-Cell Interaction<a class="anchor" aria-label="anchor" href="#options-simulation---spatial-cell-cell-interaction"></a>
</h2>
<p>The simulation of cell-cell interaction can be enabled by passing a
<code>list</code> as the <code>cci</code> option. In this list, you can
specify the following options:</p>
<div class="section level3">
<h3 id="grid-size">grid.size<a class="anchor" aria-label="anchor" href="#grid-size"></a>
</h3>
<blockquote>
<p>integer</p>
</blockquote>
<p>Manually specify the width and height of the grid.</p>
</div>
<div class="section level3">
<h3 id="layout">layout<a class="anchor" aria-label="anchor" href="#layout"></a>
</h3>
<blockquote>
<p>“enhanced”, “layers”, “islands”, or a function (default:
<code>"enhanced"</code>)</p>
</blockquote>
<p>Specify the layout of the cell types. scMultiSim provides three
built-in layouts: <code>"enhanced"</code>, <code>"layers"</code>, and
<code>"islands"</code>.</p>
<p>If set to <code>"islands"</code>, you can specify which cell types
are the islands, e.g. <code>"islands:1,2"</code>.</p>
<p>If using a custom function, it should take two arguments:
<code>function (grid_size, cell_types)</code> - grid_size: (integer) The
width and height of the grid. - cell_types: (integer vector) Each cell’s
cell type.</p>
<p>It should return a <code>n_cell x 2</code> matrix, where each row is
the x and y coordinates of a cell.</p>
</div>
<div class="section level3">
<h3 id="step-size">step.size<a class="anchor" aria-label="anchor" href="#step-size"></a>
</h3>
<blockquote>
<p>number</p>
</blockquote>
<p>If using continuous population, use this step size to further divide
the cell types on the tree. For example, if the tree only has one branch
<code>a -&gt; b</code> and the branch length is 1 while the step size is
0.34, there will be totally three cell types: a_b_1, a_b_2, a_b_3.</p>
</div>
<div class="section level3">
<h3 id="params">params<a class="anchor" aria-label="anchor" href="#params"></a>
</h3>
<blockquote>
<p>data.frame</p>
</blockquote>
<p>The spatial effect between a ligand and a receptor gene. It should be
a data frame similar to the GRN parameter, i.e. with columns
<code>receptor</code>, <code>ligand</code>, and <code>effect</code>.</p>
<p>Example:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cci</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    target    <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,   <span class="fl">6</span>,   <span class="fl">10</span>,   <span class="fl">8</span>, <span class="fl">20</span>,  <span class="fl">30</span><span class="op">)</span>,</span>
<span>    regulator <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">101</span>, <span class="fl">102</span>, <span class="fl">103</span>, <span class="fl">104</span>, <span class="fl">105</span>, <span class="fl">106</span><span class="op">)</span>,</span>
<span>    effect    <span class="op">=</span> <span class="fl">20</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="cell-type-interaction">cell.type.interaction<a class="anchor" aria-label="anchor" href="#cell-type-interaction"></a>
</h3>
<blockquote>
<p>“random” or a matrix</p>
</blockquote>
<p>Specify which cell types can communicate using which ligand-receptor
pair. It should be a 3d
<code>n_cell_types x n_cell_types x n_ligand_pair</code> numeric matrix.
The value at (i, j, k) is 1 if there exist CCI of LR-pair k between cell
type i and cell type j.</p>
<p>This matrix can be generated using the
<code><a href="../reference/cci_cell_type_params.html">cci_cell_type_params()</a></code> function. It can fill the matrix
randomly, or return an empty matrix for you to fill manually. If you
want to fill it randomly, you can simply supply <code>"random"</code>
for this option.</p>
</div>
<div class="section level3">
<h3 id="cell-type-lr-pairs">cell.type.lr.pairs<a class="anchor" aria-label="anchor" href="#cell-type-lr-pairs"></a>
</h3>
<blockquote>
<p>integer vector</p>
</blockquote>
<p>If <code>cell.type.interaction</code> is <code>"random"</code>,
specify how many LR pairs should be enabled between each cell type pair.
Should be a range, e.g. <code>4:6</code>. The actual number of LR pairs
will be uniformly sampled from this range.</p>
</div>
<div class="section level3">
<h3 id="max-neighbors">max.neighbors<a class="anchor" aria-label="anchor" href="#max-neighbors"></a>
</h3>
<blockquote>
<p>integer</p>
</blockquote>
<p>The number of interacting cells for each cell. If the cell’s
available neighbor count is not large enough, the actual interacting
cells may be smaller than this value.</p>
</div>
<div class="section level3">
<h3 id="radius">radius<a class="anchor" aria-label="anchor" href="#radius"></a>
</h3>
<blockquote>
<p>number (default: <code>1</code>), or “gaussian:sigma”</p>
</blockquote>
<p>Which cells should be considered as neighbors. The interacting cells
are those within these neighbors.</p>
<p>When it is a number, it controls the maximum distance between two
cells for them to interact.</p>
<p>When it is a string, it should be in the format
<code>gaussian:sigma</code>, for example, <code>gaussian:1.2</code>. In
this case, the probability of two cells interacting is proportional to
the distance with a Gaussian kernel applied.</p>
</div>
<div class="section level3">
<h3 id="start-layer">start.layer<a class="anchor" aria-label="anchor" href="#start-layer"></a>
</h3>
<blockquote>
<p>integer</p>
</blockquote>
<p>From which layer (time step) the simulation should start. If set to
1, the simulation will start with one cell in the grid and add one more
cell in each following layer. If set to <code>num_cells</code>, the
simulation will start from all cells available in the grid and only
continues for a few static layers, which will greatly speed up the
simulation.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hechen Li, Xiuwei Zhang, Ziqi Zhang, Michael Squires.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
