<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="scMultiSim">
<title>Available Options • scMultiSim</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Available Options">
<meta property="og:description" content="scMultiSim">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scMultiSim</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/workflow.html">Basic Workflow</a>
    <a class="dropdown-item" href="../articles/basics.html">Simulating Multimodal Single-cell Datasets</a>
    <a class="dropdown-item" href="../articles/spatialCCI.html">Simulating Spatial Cell-Cell Interactions</a>
    <a class="dropdown-item" href="../articles/options.html">Available Options</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ZhangLabGT/scMultiSim/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Available Options</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ZhangLabGT/scMultiSim/blob/HEAD/vignettes/articles/options.Rmd" class="external-link"><code>vignettes/articles/options.Rmd</code></a></small>
      <div class="d-none name"><code>options.Rmd</code></div>
    </div>

    
    
<p>This article introduces the available options in
<code>scMultiSim</code>.</p>
<div class="section level2">
<h2 id="options-general">Options: General<a class="anchor" aria-label="anchor" href="#options-general"></a>
</h2>
<div class="section level3">
<h3 id="rand-seed">rand.seed<a class="anchor" aria-label="anchor" href="#rand-seed"></a>
</h3>
<blockquote>
<p>integer (default: <code>0</code>)</p>
</blockquote>
<p>scMultiSim should produce the same result if all other parameters are
the same.</p>
</div>
<div class="section level3">
<h3 id="threads">threads<a class="anchor" aria-label="anchor" href="#threads"></a>
</h3>
<blockquote>
<p>integer (default: <code>1</code>)</p>
</blockquote>
<p>Use multithreading only when generating the CIF matrix. It will not
speed up the simulation a lot, thus not recommended.</p>
</div>
</div>
<div class="section level2">
<h2 id="options-genes">Options: Genes<a class="anchor" aria-label="anchor" href="#options-genes"></a>
</h2>
<div class="section level3">
<h3 id="grn">GRN<a class="anchor" aria-label="anchor" href="#grn"></a>
</h3>
<blockquote>
<p>A data frame with 3 columns as below. Supply <code>NA</code> to
disable the GRN effect. (required)</p>
</blockquote>
<table class="table">
<thead><tr class="header">
<th>Column</th>
<th>Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>target gene ID: <code>integer or character</code>;</td>
</tr>
<tr class="even">
<td>2</td>
<td>regulator gene ID: <code>integer or character</code>;</td>
</tr>
<tr class="odd">
<td>3</td>
<td>effect: <code>number</code>.</td>
</tr>
</tbody>
</table>
<p>If <code>num.genes</code> presents, the gene IDs should not exceed
this number. The gene IDs should start from 1 and should not ship any
intermidiate numbers.</p>
<p>Two sample datasets <code>GRN_params_100</code> and
<code>GRN_params_1000</code> from <a href="https://doi.org/10.1016/j.cels.2020.08.003" class="external-link">Dibaeinia, P., &amp;
Sinha, S. (2020)</a> are provided for testing and inspection.</p>
</div>
<div class="section level3">
<h3 id="num-genes">num.genes<a class="anchor" aria-label="anchor" href="#num-genes"></a>
</h3>
<blockquote>
<p>integer (default: <code>NULL</code>)</p>
</blockquote>
<p>If a GRN is supplied, override the total number of genes. It should
be larger than the largest gene ID in the GRN. Otherwise, the number of
genes will be determined by <code>N_genes * (1 + r_u)</code>, where
<code>r_u</code> is <code>unregulated.gene.ratio</code>.</p>
<p>If GRN is disabled, this option specifies the total number of
genes.</p>
</div>
<div class="section level3">
<h3 id="unregulated-gene-ratio">unregulated.gene.ratio<a class="anchor" aria-label="anchor" href="#unregulated-gene-ratio"></a>
</h3>
<blockquote>
<p>number &gt; 0 (default: <code>0.1</code>)</p>
</blockquote>
<p>Ratio of unreulated to regulated genes. When a GRN is supplied with
<code>N</code> genes, scMultiSim will simulate <code>N * r_u</code>
extra (unregulated) genes.</p>
</div>
<div class="section level3">
<h3 id="giv-mean-giv-sd-giv-prob">giv.mean, giv.sd, giv.prob<a class="anchor" aria-label="anchor" href="#giv-mean-giv-sd-giv-prob"></a>
</h3>
<blockquote>
<p>(default: <code>0, 1, 0.3</code>)</p>
</blockquote>
<p>The parameters used to sample the GIV matrix. With probability
<code>giv.prob</code>, the value is sampled from
N(<code>giv.mean</code>, <code>giv.sd</code>). Otherwise the value is
0.</p>
</div>
<div class="section level3">
<h3 id="dynamic-grn">dynamic.GRN<a class="anchor" aria-label="anchor" href="#dynamic-grn"></a>
</h3>
<blockquote>
<p>list (default: <code>NULL</code>)</p>
</blockquote>
<p>Enables dynamic (cell-specific GRN). Run
<code>scmultisim_help("dynamic.GRN")</code> to see more
explaination.</p>
</div>
<div class="section level3">
<h3 id="hge-prop-hge-mean-hge-sd">hge.prop, hge.mean, hge.sd<a class="anchor" aria-label="anchor" href="#hge-prop-hge-mean-hge-sd"></a>
</h3>
<blockquote>
<p>(default: <code>0, 5, 1</code>)</p>
</blockquote>
<p>Treat some random genes as highly-expressed (house-keeping) genes. A
proportion of <code>hge.prop</code> genes will have expression scaled by
a multiplier sampled from N(<code>hge.mean</code>,
<code>hge.sd</code>).</p>
</div>
<div class="section level3">
<h3 id="hge-range">hge.range<a class="anchor" aria-label="anchor" href="#hge-range"></a>
</h3>
<blockquote>
<p>integer (default: <code>1</code>)</p>
</blockquote>
<p>When selecting highly-expressed genes, only choose genes with ID &gt;
<code>hge.range</code>.</p>
</div>
<div class="section level3">
<h3 id="hge-max-var">hge.max.var<a class="anchor" aria-label="anchor" href="#hge-max-var"></a>
</h3>
<blockquote>
<p>number (default: <code>500</code>)</p>
</blockquote>
<p>When selecting highly-expressed genes, only choose genes with
variation &lt; <code>hge.max.var</code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="options-cells">Options: Cells<a class="anchor" aria-label="anchor" href="#options-cells"></a>
</h2>
<div class="section level3">
<h3 id="num-cells">num.cells<a class="anchor" aria-label="anchor" href="#num-cells"></a>
</h3>
<blockquote>
<p>integer (default: <code>1000</code>)</p>
</blockquote>
<p>The number of cells to be simulated.</p>
</div>
<div class="section level3">
<h3 id="tree">tree<a class="anchor" aria-label="anchor" href="#tree"></a>
</h3>
<blockquote>
<p>phylo (default: <code><a href="../reference/Phyla5.html">Phyla5()</a></code>)</p>
</blockquote>
<p>The cell differential tree, which will be used to generate cell
trajectories (if <code>discrete.cif = T</code>) or clusters (if
<code>discrete.cif = F</code>). In discrete population mode, only the
tree tips will be used. Three demo trees, <code><a href="../reference/Phyla5.html">Phyla5()</a></code>,
<code><a href="../reference/Phyla3.html">Phyla3()</a></code> and <code><a href="../reference/Phyla1.html">Phyla1()</a></code>, are provided.</p>
</div>
<div class="section level3">
<h3 id="discrete-cif">discrete.cif<a class="anchor" aria-label="anchor" href="#discrete-cif"></a>
</h3>
<blockquote>
<p>logical (default: <code>FALSE</code>)</p>
</blockquote>
<p>Whether the cell population is discrete (continuous otherwise).</p>
</div>
<div class="section level3">
<h3 id="discrete-min-pop-size-discrete-min-pop-index">discrete.min.pop.size, discrete.min.pop.index<a class="anchor" aria-label="anchor" href="#discrete-min-pop-size-discrete-min-pop-index"></a>
</h3>
<blockquote>
<p>integer, integer (default: <code>70, 1</code>)</p>
</blockquote>
<p>In discrete population mode, specify one cluster to have the smallest
cell population. The cluster will contain
<code>discrete.min.pop.size</code> cells.
<code>discrete.min.pop.index</code> should be a valid cluster index
(tree tip number).</p>
</div>
<div class="section level3">
<h3 id="discrete-pop-size">discrete.pop.size<a class="anchor" aria-label="anchor" href="#discrete-pop-size"></a>
</h3>
<blockquote>
<p>integer vector (default: <code>NA</code>);
e.g. <code>c(200, 250, 300)</code></p>
</blockquote>
<p>Manually specify the size of each cluster.</p>
</div>
</div>
<div class="section level2">
<h2 id="options-cif">Options: CIF<a class="anchor" aria-label="anchor" href="#options-cif"></a>
</h2>
<div class="section level3">
<h3 id="num-cifs">num.cifs<a class="anchor" aria-label="anchor" href="#num-cifs"></a>
</h3>
<blockquote>
<p>integer (default: <code>50</code>)</p>
</blockquote>
<p>Total number of differential and non-differential CIFs, which can be
viewed as latent representation of cells.</p>
</div>
<div class="section level3">
<h3 id="diff-cif-fraction">diff.cif.fraction<a class="anchor" aria-label="anchor" href="#diff-cif-fraction"></a>
</h3>
<blockquote>
<p>number (default: <code>0.9</code>)</p>
</blockquote>
<p>Fraction of differential CIFs. Differential CIFs encode the cell type
information, while non-differential CIFs are randomly sampled for each
cell.</p>
</div>
<div class="section level3">
<h3 id="cif-center-cif-sigma">cif.center, cif.sigma<a class="anchor" aria-label="anchor" href="#cif-center-cif-sigma"></a>
</h3>
<blockquote>
<p>(default: <code>1, 0.1</code>)</p>
</blockquote>
<p>The distribution used to sample CIF values.</p>
</div>
<div class="section level3">
<h3 id="use-impulse">use.impulse<a class="anchor" aria-label="anchor" href="#use-impulse"></a>
</h3>
<blockquote>
<p>logical (default: <code>FALSE</code>)</p>
</blockquote>
<p>In continuous population mode, when sampling CIFs along the tree, use
the impulse model rather than the default gaussian random walk.</p>
</div>
</div>
<div class="section level2">
<h2 id="options-simulation---atac">Options: Simulation - ATAC<a class="anchor" aria-label="anchor" href="#options-simulation---atac"></a>
</h2>
<div class="section level3">
<h3 id="atac-effect">atac.effect<a class="anchor" aria-label="anchor" href="#atac-effect"></a>
</h3>
<blockquote>
<p>number ∈ [0, 1] (default: <code>0.5</code>)</p>
</blockquote>
<p>The influence of chromatin accessability data on gene expression.</p>
</div>
<div class="section level3">
<h3 id="region-distrib">region.distrib<a class="anchor" aria-label="anchor" href="#region-distrib"></a>
</h3>
<blockquote>
<p>vector of length 3, should sum to 1 (default:
<code>c(0.1, 0.5, 0.4)</code>)</p>
</blockquote>
<p>The probability that a gene is regulated by 0, 1, 2 consecutive
regions, respectively.</p>
</div>
<div class="section level3">
<h3 id="atac-p_zero">atac.p_zero<a class="anchor" aria-label="anchor" href="#atac-p_zero"></a>
</h3>
<blockquote>
<p>number ∈ [0, 1] (default: <code>0.8</code>)</p>
</blockquote>
<p>The proportion of zeros we see in the simulated scATAC-seq data.</p>
</div>
<div class="section level3">
<h3 id="riv-mean-riv-sd-riv-prob">riv.mean, riv.sd, riv.prob<a class="anchor" aria-label="anchor" href="#riv-mean-riv-sd-riv-prob"></a>
</h3>
<blockquote>
<p>(default: <code>0, 1, 0.3</code>)</p>
</blockquote>
<p>The parameters used to sample the RIV (Region Identity Vectors). With
probability <code>riv.prob</code>, the value is sampled from
N(<code>riv.mean</code>, <code>riv.sd</code>). Otherwise the value is
0.</p>
</div>
</div>
<div class="section level2">
<h2 id="options-simulation---rna">Options: Simulation - RNA<a class="anchor" aria-label="anchor" href="#options-simulation---rna"></a>
</h2>
<div class="section level3">
<h3 id="do-velocity">do.velocity<a class="anchor" aria-label="anchor" href="#do-velocity"></a>
</h3>
<blockquote>
<p>logical (default: <code>FALSE</code>)</p>
</blockquote>
<p>When set to <code>TRUE</code>, simulate using the full kinetic model
and generate RNA velocity data. Otherwise, the Beta-Poission model will
be used.</p>
</div>
<div class="section level3">
<h3 id="beta">beta<a class="anchor" aria-label="anchor" href="#beta"></a>
</h3>
<blockquote>
<p>number (default: <code>0.4</code>)</p>
</blockquote>
<p>The splicing rate of each gene in the kinetic model.</p>
</div>
<div class="section level3">
<h3 id="d">d<a class="anchor" aria-label="anchor" href="#d"></a>
</h3>
<blockquote>
<p>number (default: <code>1</code>)</p>
</blockquote>
<p>The degradation rate of each gene in the kinetic model.</p>
</div>
<div class="section level3">
<h3 id="num-cycles">num.cycles<a class="anchor" aria-label="anchor" href="#num-cycles"></a>
</h3>
<blockquote>
<p>number (default: <code>3</code>)</p>
</blockquote>
<p>The number of cycles run before sampling the gene expression of a
cell.</p>
</div>
<div class="section level3">
<h3 id="cycle-len">cycle.len<a class="anchor" aria-label="anchor" href="#cycle-len"></a>
</h3>
<blockquote>
<p>number (default: <code>1</code>)</p>
</blockquote>
<p>In velocity mode, a multiplier for the cell cycle length. It is
multiplied by the expected time to transition from k_on to k_off and
back to form the the length of a cycle.</p>
</div>
</div>
<div class="section level2">
<h2 id="options-simulation---spatial-cell-cell-interaction">Options: Simulation - Spatial Cell-Cell Interaction<a class="anchor" aria-label="anchor" href="#options-simulation---spatial-cell-cell-interaction"></a>
</h2>
<div class="section level3">
<h3 id="cci">cci<a class="anchor" aria-label="anchor" href="#cci"></a>
</h3>
<p>Enables cell-cell interaction. See
<code>scmultisim_help("cci")</code> for details.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hechen Li, Xiuwei Zhang, Michael Squires.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
