<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="scMultiSim">
<title>Basic Workflow • scMultiSim</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Basic Workflow">
<meta property="og:description" content="scMultiSim">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scMultiSim</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/workflow.html">Basic Workflow</a>
    <a class="dropdown-item" href="../articles/basics.html">Simulating Multimodal Single-cell Datasets</a>
    <a class="dropdown-item" href="../articles/spatialCCI.html">Simulating Spatial Cell-Cell Interactions</a>
    <a class="dropdown-item" href="../articles/options.html">Available Options</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ZhangLabGT/scMultiSim/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Basic Workflow</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ZhangLabGT/scMultiSim/blob/HEAD/vignettes/articles/workflow.Rmd" class="external-link"><code>vignettes/articles/workflow.Rmd</code></a></small>
      <div class="d-none name"><code>workflow.Rmd</code></div>
    </div>

    
    
<p>This article introduces the basic workflow of
<code>scMultiSim</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://zhanglabgt.github.io/scMultiSim/">scMultiSim</a></span><span class="op">)</span></span></code></pre></div>
<p>A typical workflow consists these three main steps:</p>
<ol style="list-style-type: decimal">
<li>Simulate the true counts;</li>
<li>Add technical noise to the dataset;</li>
<li>Add batch effects.</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/divide_batches.html">divide_batches</a></span><span class="op">(</span><span class="va">results</span>, nbatch <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="step-1-simulate-true-counts">Step 1: Simulate true counts<a class="anchor" aria-label="anchor" href="#step-1-simulate-true-counts"></a>
</h2>
<p>Call the <code>sim_true_counts</code> function to simulate the true
counts. It accepts a list of options as input. This
<code>sim_true_counts</code> is the most important step, and you are
able to control most of the simulated effects here.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GRN_params_100</span><span class="op">)</span></span>
<span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sim_true_counts.html">sim_true_counts</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="co"># required options</span></span>
<span>  GRN <span class="op">=</span> <span class="va">GRN_params_100</span>,</span>
<span>  tree <span class="op">=</span> <span class="fu"><a href="../reference/Phyla5.html">Phyla5</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  num.cells <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  <span class="co"># optional options</span></span>
<span>  num.cif <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  discrete.cif <span class="op">=</span> <span class="cn">F</span>,</span>
<span>  cif.sigma <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  do.velocity <span class="op">=</span> <span class="cn">T</span></span>
<span>  <span class="co"># ... other options</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Time spent: 0.14 mins</span></span></code></pre></div>
<p>scMultiSim requires users to provide the following options:</p>
<ul>
<li>
<a href="#grn"><code>GRN</code></a>: The Gene Regulatory Network. If
the GRN effect is unneeded, Provide <code>NA</code>.</li>
<li>
<a href="#tree"><code>tree</code></a>: The cell differential tree.
If the cell trajectory is unimportant, simply use <code><a href="../reference/Phyla1.html">Phyla1()</a></code>
to generate a linear tree.</li>
</ul>
<p>scMultiSim provides several example GRN and trajectory data: - GRN:
<code>GRN_params_100</code> and <code>GRN_params_1139</code> - Tree:
<code><a href="../reference/Phyla5.html">Phyla5()</a></code>, <code><a href="../reference/Phyla3.html">Phyla3()</a></code> and
<code><a href="../reference/Phyla1.html">Phyla1()</a></code></p>
<p>You can visualize these data as below:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_grn.html">plot_grn</a></span><span class="op">(</span><span class="va">GRN_params_100</span><span class="op">)</span></span></code></pre></div>
<p><img src="workflow_files/figure-html/unnamed-chunk-2-1.png" width="480"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_phyla.html">plot_phyla</a></span><span class="op">(</span><span class="fu"><a href="../reference/Phyla5.html">Phyla5</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="workflow_files/figure-html/unnamed-chunk-2-2.png" width="480"></p>
<p>Typically, you may also want to adjust the following options to
control the basic cell population:</p>
<ul>
<li>
<a href="#numcells"><code>num.cells</code></a>: Specify the number
of cells.</li>
<li>
<a href="#unregulatedgeneratio"><code>unregulated.gene.ratio</code></a> or
<a href="#numgenes"><code>num.genes</code></a>: Control the total number
of genes.</li>
<li>
<a href="#discretecif"><code>discrete.cif</code></a>: Whether
generating discrete or continuous cell population.</li>
<li>
<a href="#diffciffraction"><code>diff.cif.fraction</code></a>:
Control the contribution of the trajectory/cluster specified by the
tree.</li>
<li>
<a href="#cifcenter-cifsigma"><code>cif.sigma</code></a>: Control
the variation of cells along the trajectory.</li>
</ul>
<div class="section level3">
<h3 id="effects-of-some-common-options">Effects of some common options<a class="anchor" aria-label="anchor" href="#effects-of-some-common-options"></a>
</h3>
<p>Here, we provide a quick guide on adjusting the effect of each
biological factors:</p>
<div class="section level5">
<h5 id="diff-cif-fraction">
<code>diff.cif.fraction</code><a class="anchor" aria-label="anchor" href="#diff-cif-fraction"></a>
</h5>
<p>Controls the balance between trajectory and GRN.</p>
<table class="table">
<colgroup>
<col width="45%">
<col width="54%">
</colgroup>
<thead><tr class="header">
<th>Higher Value</th>
<th>Lower Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<strong>Clear</strong> trajectories / cluster boundaries</td>
<td>
<strong>Vague</strong> trajectories / cluster boundaries</td>
</tr>
<tr class="even">
<td>Trajectory determined by the tree</td>
<td>Trajectory determined by other factors, e.g., GRN</td>
</tr>
</tbody>
</table>
</div>
<div class="section level5">
<h5 id="cif-sigma">
<code>cif.sigma</code><a class="anchor" aria-label="anchor" href="#cif-sigma"></a>
</h5>
<p>Controls the noise along trajectory.</p>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Higher Value</th>
<th>Lower Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<strong>Vague</strong> trajectories / cluster boundaries</td>
<td>
<strong>Clear</strong> trajectories / cluster boundaries</td>
</tr>
<tr class="even">
<td>Cells stick to the trajectory</td>
<td>More randomness in the trajectory</td>
</tr>
</tbody>
</table>
</div>
<div class="section level5">
<h5 id="intrisic-noise">
<code>intrisic.noise</code><a class="anchor" aria-label="anchor" href="#intrisic-noise"></a>
</h5>
<p>Controls the randomness added during sampling mRNA counts, therefore
also affecting the trajectory shape.</p>
<table class="table">
<colgroup>
<col width="45%">
<col width="55%">
</colgroup>
<thead><tr class="header">
<th>Higher Value</th>
<th>Lower Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<strong>Clear</strong> trajectories / cluster boundaries</td>
<td>
<strong>Vague</strong> trajectories / cluster boundaries</td>
</tr>
<tr class="even">
<td>Expression contains more noise</td>
<td>Expression directly computed from kinectic parameters</td>
</tr>
</tbody>
</table>
</div>
<div class="section level5">
<h5 id="atac-effect">
<code>atac.effect</code><a class="anchor" aria-label="anchor" href="#atac-effect"></a>
</h5>
<p>Controls the contribution of the chtomatin accessibility.</p>
<table class="table">
<colgroup>
<col width="50%">
<col width="49%">
</colgroup>
<thead><tr class="header">
<th>Higher Value</th>
<th>Lower Value</th>
</tr></thead>
<tbody><tr class="odd">
<td>Higher correlation between ATAC and RNA-seq</td>
<td>Lower correlation between ATAC and RNA-seq</td>
</tr></tbody>
</table>
<p>You can find more information about the options in the <em>Available
Options</em> article.</p>
</div>
</div>
<div class="section level3">
<h3 id="the-shiny-app">The Shiny app<a class="anchor" aria-label="anchor" href="#the-shiny-app"></a>
</h3>
<p>Don’t forget that scMultiSim provides a Shiny app to help you explore
the options interactively. Simply run <code><a href="../reference/run_shiny.html">run_shiny()</a></code> to start
the app.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/run_shiny.html">run_shiny</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="return-value">Return value<a class="anchor" aria-label="anchor" href="#return-value"></a>
</h3>
<p>The return value will be a <code>scMultiSim Environment</code>
object, and you can access various data and parameters using the
<code>$</code> operator.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] ".grn"             "unspliced_counts" ".options"         ".n"              </span></span>
<span><span class="co">#&gt;  [5] "region_to_gene"   "atacseq_data"     "giv"              "kinetic_params"  </span></span>
<span><span class="co">#&gt;  [9] "num_genes"        "velocity"         "cif"              "hge_scale"       </span></span>
<span><span class="co">#&gt; [13] "counts"           "cell_meta"        "atac_counts"      "region_to_tf"    </span></span>
<span><span class="co">#&gt; [17] "grn_params"       "cell_time"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 110 500</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="step-2-add-technical-noise">Step 2: Add technical noise<a class="anchor" aria-label="anchor" href="#step-2-add-technical-noise"></a>
</h2>
<p>Simply use the <code>add_expr_noise</code> function to add technical
noise to the dataset.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/add_expr_noise.html">add_expr_noise</a></span><span class="op">(</span></span>
<span>  <span class="va">results</span>,</span>
<span>  <span class="co"># options go here</span></span>
<span>  alpha_mean <span class="op">=</span> <span class="fl">1e4</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Adding experimental noise...</span></span>
<span><span class="co">#&gt; 50..100..150..200..250..300..350..400..450..500..Using atac_counts</span></span>
<span><span class="co">#&gt; Time spent: 0.41 mins</span></span></code></pre></div>
<p>A <code>counts_obs</code> field will be added to the
<code>results</code> object.</p>
<p>This function also accepts a list of options. See the documentation
for more details.</p>
<ul>
<li>
<code>protocol</code>: <code>"umi"</code> or <code>"nonUMI"</code>,
whether simulate the UMI protocol.</li>
<li>
<code>alpha_mean</code>, <code>alpha_sd</code>: Mean and deviation
of rate of subsampling of transcripts during capture step.</li>
<li>
<code>alpha_gene_mean</code>, <code>alpha_gene_sd</code>:
<code>alpha</code> parameters, but gene-wise.</li>
<li>
<code>depth_mean</code>, <code>depth_sd</code>: Mean and deviation
of sequencing depth.</li>
<li>
<code>gene_len</code>: A vector with lengths of all genes.</li>
<li>
<code>atac.obs.prob</code>: For each integer count of a particular
region for a particular cell, the probability the count will be
observed.</li>
<li>
<code>atac.sd.frac</code>: The fraction of ATAC-seq data value used
as the standard deviation of added normally distrubted noise.</li>
<li>
<code>randseed</code>: random seed.</li>
</ul>
</div>
<div class="section level2">
<h2 id="step-3-add-batch-effects">Step 3: Add batch effects<a class="anchor" aria-label="anchor" href="#step-3-add-batch-effects"></a>
</h2>
<p>Finally, use the <code>divide_batches</code> function to add batch
effects.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/divide_batches.html">divide_batches</a></span><span class="op">(</span></span>
<span>  <span class="va">results</span>,</span>
<span>  nbatch <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  effect <span class="op">=</span> <span class="fl">0.5</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Adding batch effects...</span></span></code></pre></div>
<p>A <code>counts_with_batches</code> field will be added to the
<code>results</code> object.</p>
<p>The available options are:</p>
<ul>
<li>
<code>nbatch</code>: Number of batches.</li>
<li>
<code>effect</code>: The batch effect size.</li>
</ul>
</div>
<div class="section level2">
<h2 id="data-available-in-the-result">Data available in the result<a class="anchor" aria-label="anchor" href="#data-available-in-the-result"></a>
</h2>
<p>scMultiSim returns an environment with the following fields:</p>
<ul>
<li>
<code>counts</code>: Gene-by-cell scRNA-seq counts.</li>
<li>
<code>atac_counts</code>: Region-by-cell scATAC-seq counts.</li>
<li>
<code>region_to_gene</code>: Region-by-gene 0-1 marix indicating the
corresponding relationship between chtomatin regions and genes.</li>
<li>
<code>atacseq_data</code>: The “clean” scATAC-seq counts without
added intrinsic noise.</li>
<li>
<code>cell_meta</code>: A dataframe containing cell type labels and
pseudotime information.</li>
<li>
<code>cif</code>: The CIF used during the simulation.</li>
<li>
<code>giv</code>: The GIV used during the simulation.</li>
<li>
<code>kinetic_params</code>: The kinetic parameters used during the
simulation.</li>
<li>
<code>.grn</code>: The GRN used during the simulation.</li>
<li>
<code>.grn$regulators</code>: The list of TFs used by all gene-by-TF
matrices.</li>
<li>
<code>.grn$geff</code>: Gene-by-TF matrix representing the GRN used
during the simulation.</li>
<li>
<code>.n</code>: Other metadata, e.g. <code>.n$cells</code> is the
number of cells.</li>
</ul>
<p>If <code>do.velocity</code> is enabled, it has these additional
fields:</p>
<ul>
<li>
<code>unspliced_counts</code>: Gene-by-cell unspliced RNA
counts.</li>
<li>
<code>velocity</code>: Gene-by-cell RNA velocity ground truth.</li>
<li>
<code>cell_time</code>: The pseudotime at which the cell counts were
generated.</li>
</ul>
<p>If dynamic GRN is enabled, it has these additional fields:</p>
<ul>
<li>
<code>cell_specific_grn</code>: A list of length
<code>n_cells</code>. Each element is a gene-by-TF matrix, indicating
the cell’s GRN.</li>
</ul>
<p>If cell-cell interaction is enabled, it has these additional
fields:</p>
<ul>
<li>
<code>grid</code>: The grid object used during the simulation.</li>
<li>
<code>grid$get_neighbours(i)</code>: Get the neighbour cells of cell
<code>i</code>.</li>
<li>
<code>cci_locs</code>: A dataframe containing the X and Y
coordinates of each cell.</li>
<li>
<code>cci_cell_type_param</code>: A dataframe containing the CCI
network ground truth: all ligand-receptor pairs between each pair of
cell types.</li>
<li>
<code>cci_cell_types</code>: For continuous cell population, the
sub-divided cell types along the trajectory used when simulating
CCI.</li>
</ul>
<p>If it is a debug session (<code>debug = TRUE</code>), a
<code>sim</code> field is available, which is an environment contains
all internal states and data structures.</p>
</div>
<div class="section level2">
<h2 id="visualize-the-results">Visualize the results<a class="anchor" aria-label="anchor" href="#visualize-the-results"></a>
</h2>
<p>scMultiSim provides various visualization functions to help you
understand the simulated data.</p>
<p><code>plot_tsne</code> visualizes the cells using t-SNE.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_tsne.html">plot_tsne</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">counts</span>, <span class="va">results</span><span class="op">$</span><span class="va">cell_meta</span><span class="op">$</span><span class="va">pop</span><span class="op">)</span></span></code></pre></div>
<p><img src="workflow_files/figure-html/visualize-1.png" width="700"></p>
<p>There are more visualization functions available in the package, such
as:</p>
<ul>
<li>
<code>plot_gene_module_cor_heatmap</code>: Plot the correlation
heatmap of genes.</li>
<li>
<code>plot_rna_velocity</code>: Plot the RNA velocity.</li>
<li>
<code>plot_cell_loc</code>: Plot the cell locations in simulated
spatial data.</li>
</ul>
<p>Please check out the reference page for more information.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hechen Li, Xiuwei Zhang, Michael Squires.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
